// react-beautiful-color/dist/index.mjs
import ar, { useRef as ve, useCallback as A, useEffect as Pe, useMemo as we, useState as _e, createContext as lr, useContext as ir } from "react";
var he = { exports: {} }, ce = {};
/**
 * @license React
 * react-jsx-runtime.production.js
 *
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var He;
function cr() {
  if (He) return ce;
  He = 1;
  var e = Symbol.for("react.transitional.element"), r = Symbol.for("react.fragment");
  function t(o, n, a) {
    var s = null;
    if (a !== void 0 && (s = "" + a), n.key !== void 0 && (s = "" + n.key), "key" in n) {
      a = {};
      for (var c in n)
        c !== "key" && (a[c] = n[c]);
    } else a = n;
    return n = a.ref, {
      $$typeof: e,
      type: o,
      key: s,
      ref: n !== void 0 ? n : null,
      props: a
    };
  }
  return ce.Fragment = r, ce.jsx = t, ce.jsxs = t, ce;
}
var de = {};
/**
 * @license React
 * react-jsx-runtime.development.js
 *
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var Ie;
function dr() {
  return Ie || (Ie = 1, process.env.NODE_ENV !== "production" && (function() {
    function e(l) {
      if (l == null) return null;
      if (typeof l == "function")
        return l.$$typeof === me ? null : l.displayName || l.name || null;
      if (typeof l == "string") return l;
      switch (l) {
        case S:
          return "Fragment";
        case H:
          return "Profiler";
        case j:
          return "StrictMode";
        case V:
          return "Suspense";
        case I:
          return "SuspenseList";
        case N:
          return "Activity";
      }
      if (typeof l == "object")
        switch (typeof l.tag == "number" && console.error(
          "Received an unexpected object in getComponentNameFromType(). This is likely a bug in React. Please file an issue."
        ), l.$$typeof) {
          case C:
            return "Portal";
          case J:
            return (l.displayName || "Context") + ".Provider";
          case re:
            return (l._context.displayName || "Context") + ".Consumer";
          case D:
            var h = l.render;
            return l = l.displayName, l || (l = h.displayName || h.name || "", l = l !== "" ? "ForwardRef(" + l + ")" : "ForwardRef"), l;
          case X:
            return h = l.displayName || null, h !== null ? h : e(l.type) || "Memo";
          case p:
            h = l._payload, l = l._init;
            try {
              return e(l(h));
            } catch {
            }
        }
      return null;
    }
    function r(l) {
      return "" + l;
    }
    function t(l) {
      try {
        r(l);
        var h = !1;
      } catch {
        h = !0;
      }
      if (h) {
        h = console;
        var v = h.error, x = typeof Symbol == "function" && Symbol.toStringTag && l[Symbol.toStringTag] || l.constructor.name || "Object";
        return v.call(
          h,
          "The provided key is an unsupported type %s. This value must be coerced to a string before using it here.",
          x
        ), r(l);
      }
    }
    function o(l) {
      if (l === S) return "<>";
      if (typeof l == "object" && l !== null && l.$$typeof === p)
        return "<...>";
      try {
        var h = e(l);
        return h ? "<" + h + ">" : "<...>";
      } catch {
        return "<...>";
      }
    }
    function n() {
      var l = te.A;
      return l === null ? null : l.getOwner();
    }
    function a() {
      return Error("react-stack-top-frame");
    }
    function s(l) {
      if (K.call(l, "key")) {
        var h = Object.getOwnPropertyDescriptor(l, "key").get;
        if (h && h.isReactWarning) return !1;
      }
      return l.key !== void 0;
    }
    function c(l, h) {
      function v() {
        L || (L = !0, console.error(
          "%s: `key` is not a prop. Trying to access it will result in `undefined` being returned. If you need to access the same value within the child component, you should pass it as a different prop. (https://react.dev/link/special-props)",
          h
        ));
      }
      v.isReactWarning = !0, Object.defineProperty(l, "key", {
        get: v,
        configurable: !0
      });
    }
    function i() {
      var l = e(this.type);
      return z[l] || (z[l] = !0, console.error(
        "Accessing element.ref was removed in React 19. ref is now a regular prop. It will be removed from the JSX Element type in a future release."
      )), l = this.props.ref, l !== void 0 ? l : null;
    }
    function m(l, h, v, x, y, _, oe, k) {
      return v = _.ref, l = {
        $$typeof: R,
        type: l,
        key: h,
        props: _,
        _owner: y
      }, (v !== void 0 ? v : null) !== null ? Object.defineProperty(l, "ref", {
        enumerable: !1,
        get: i
      }) : Object.defineProperty(l, "ref", { enumerable: !1, value: null }), l._store = {}, Object.defineProperty(l._store, "validated", {
        configurable: !1,
        enumerable: !1,
        writable: !0,
        value: 0
      }), Object.defineProperty(l, "_debugInfo", {
        configurable: !1,
        enumerable: !1,
        writable: !0,
        value: null
      }), Object.defineProperty(l, "_debugStack", {
        configurable: !1,
        enumerable: !1,
        writable: !0,
        value: oe
      }), Object.defineProperty(l, "_debugTask", {
        configurable: !1,
        enumerable: !1,
        writable: !0,
        value: k
      }), Object.freeze && (Object.freeze(l.props), Object.freeze(l)), l;
    }
    function b(l, h, v, x, y, _, oe, k) {
      var T = h.children;
      if (T !== void 0)
        if (x)
          if (fe(T)) {
            for (x = 0; x < T.length; x++)
              P(T[x]);
            Object.freeze && Object.freeze(T);
          } else
            console.error(
              "React.jsx: Static children should always be an array. You are likely explicitly calling React.jsxs or React.jsxDEV. Use the Babel transform instead."
            );
        else P(T);
      if (K.call(h, "key")) {
        T = e(l);
        var $ = Object.keys(h).filter(function(ie) {
          return ie !== "key";
        });
        x = 0 < $.length ? "{key: someKey, " + $.join(": ..., ") + ": ...}" : "{key: someKey}", le[T + x] || ($ = 0 < $.length ? "{" + $.join(": ..., ") + ": ...}" : "{}", console.error(
          `A props object containing a "key" prop is being spread into JSX:
  let props = %s;
  <%s {...props} />
React keys must be passed directly to JSX without using spread:
  let props = %s;
  <%s key={someKey} {...props} />`,
          x,
          T,
          $,
          T
        ), le[T + x] = !0);
      }
      if (T = null, v !== void 0 && (t(v), T = "" + v), s(h) && (t(h.key), T = "" + h.key), "key" in h) {
        v = {};
        for (var Y in h)
          Y !== "key" && (v[Y] = h[Y]);
      } else v = h;
      return T && c(
        v,
        typeof l == "function" ? l.displayName || l.name || "Unknown" : l
      ), m(
        l,
        T,
        _,
        y,
        n(),
        v,
        oe,
        k
      );
    }
    function P(l) {
      typeof l == "object" && l !== null && l.$$typeof === R && l._store && (l._store.validated = 1);
    }
    var w = ar, R = Symbol.for("react.transitional.element"), C = Symbol.for("react.portal"), S = Symbol.for("react.fragment"), j = Symbol.for("react.strict_mode"), H = Symbol.for("react.profiler"), re = Symbol.for("react.consumer"), J = Symbol.for("react.context"), D = Symbol.for("react.forward_ref"), V = Symbol.for("react.suspense"), I = Symbol.for("react.suspense_list"), X = Symbol.for("react.memo"), p = Symbol.for("react.lazy"), N = Symbol.for("react.activity"), me = Symbol.for("react.client.reference"), te = w.__CLIENT_INTERNALS_DO_NOT_USE_OR_WARN_USERS_THEY_CANNOT_UPGRADE, K = Object.prototype.hasOwnProperty, fe = Array.isArray, Z = console.createTask ? console.createTask : function() {
      return null;
    };
    w = {
      react_stack_bottom_frame: function(l) {
        return l();
      }
    };
    var L, z = {}, O = w.react_stack_bottom_frame.bind(
      w,
      a
    )(), f = Z(o(a)), le = {};
    de.Fragment = S, de.jsx = function(l, h, v, x, y) {
      var _ = 1e4 > te.recentlyCreatedOwnerStacks++;
      return b(
        l,
        h,
        v,
        !1,
        x,
        y,
        _ ? Error("react-stack-top-frame") : O,
        _ ? Z(o(l)) : f
      );
    }, de.jsxs = function(l, h, v, x, y) {
      var _ = 1e4 > te.recentlyCreatedOwnerStacks++;
      return b(
        l,
        h,
        v,
        !0,
        x,
        y,
        _ ? Error("react-stack-top-frame") : O,
        _ ? Z(o(l)) : f
      );
    };
  })()), de;
}
var Le;
function ur() {
  return Le || (Le = 1, process.env.NODE_ENV === "production" ? he.exports = cr() : he.exports = dr()), he.exports;
}
var E = ur();
function qe(e) {
  var r, t, o = "";
  if (typeof e == "string" || typeof e == "number") o += e;
  else if (typeof e == "object") if (Array.isArray(e)) {
    var n = e.length;
    for (r = 0; r < n; r++) e[r] && (t = qe(e[r])) && (o && (o += " "), o += t);
  } else for (t in e) e[t] && (o && (o += " "), o += t);
  return o;
}
function mr() {
  for (var e, r, t = 0, o = "", n = arguments.length; t < n; t++) (e = arguments[t]) && (r = qe(e)) && (o && (o += " "), o += r);
  return o;
}
const je = "-", fr = (e) => {
  const r = hr(e), {
    conflictingClassGroups: t,
    conflictingClassGroupModifiers: o
  } = e;
  return {
    getClassGroupId: (s) => {
      const c = s.split(je);
      return c[0] === "" && c.length !== 1 && c.shift(), Je(c, r) || pr(s);
    },
    getConflictingClassGroupIds: (s, c) => {
      const i = t[s] || [];
      return c && o[s] ? [...i, ...o[s]] : i;
    }
  };
}, Je = (e, r) => {
  if (e.length === 0)
    return r.classGroupId;
  const t = e[0], o = r.nextPart.get(t), n = o ? Je(e.slice(1), o) : void 0;
  if (n)
    return n;
  if (r.validators.length === 0)
    return;
  const a = e.join(je);
  return r.validators.find(({
    validator: s
  }) => s(a))?.classGroupId;
}, Ge = /^\[(.+)\]$/, pr = (e) => {
  if (Ge.test(e)) {
    const r = Ge.exec(e)[1], t = r?.substring(0, r.indexOf(":"));
    if (t)
      return "arbitrary.." + t;
  }
}, hr = (e) => {
  const {
    theme: r,
    classGroups: t
  } = e, o = {
    nextPart: /* @__PURE__ */ new Map(),
    validators: []
  };
  for (const n in t)
    Se(t[n], o, n, r);
  return o;
}, Se = (e, r, t, o) => {
  e.forEach((n) => {
    if (typeof n == "string") {
      const a = n === "" ? r : Fe(r, n);
      a.classGroupId = t;
      return;
    }
    if (typeof n == "function") {
      if (gr(n)) {
        Se(n(o), r, t, o);
        return;
      }
      r.validators.push({
        validator: n,
        classGroupId: t
      });
      return;
    }
    Object.entries(n).forEach(([a, s]) => {
      Se(s, Fe(r, a), t, o);
    });
  });
}, Fe = (e, r) => {
  let t = e;
  return r.split(je).forEach((o) => {
    t.nextPart.has(o) || t.nextPart.set(o, {
      nextPart: /* @__PURE__ */ new Map(),
      validators: []
    }), t = t.nextPart.get(o);
  }), t;
}, gr = (e) => e.isThemeGetter, br = (e) => {
  if (e < 1)
    return {
      get: () => {
      },
      set: () => {
      }
    };
  let r = 0, t = /* @__PURE__ */ new Map(), o = /* @__PURE__ */ new Map();
  const n = (a, s) => {
    t.set(a, s), r++, r > e && (r = 0, o = t, t = /* @__PURE__ */ new Map());
  };
  return {
    get(a) {
      let s = t.get(a);
      if (s !== void 0)
        return s;
      if ((s = o.get(a)) !== void 0)
        return n(a, s), s;
    },
    set(a, s) {
      t.has(a) ? t.set(a, s) : n(a, s);
    }
  };
}, Ce = "!", Te = ":", xr = Te.length, vr = (e) => {
  const {
    prefix: r,
    experimentalParseClassName: t
  } = e;
  let o = (n) => {
    const a = [];
    let s = 0, c = 0, i = 0, m;
    for (let C = 0; C < n.length; C++) {
      let S = n[C];
      if (s === 0 && c === 0) {
        if (S === Te) {
          a.push(n.slice(i, C)), i = C + xr;
          continue;
        }
        if (S === "/") {
          m = C;
          continue;
        }
      }
      S === "[" ? s++ : S === "]" ? s-- : S === "(" ? c++ : S === ")" && c--;
    }
    const b = a.length === 0 ? n : n.substring(i), P = wr(b), w = P !== b, R = m && m > i ? m - i : void 0;
    return {
      modifiers: a,
      hasImportantModifier: w,
      baseClassName: P,
      maybePostfixModifierPosition: R
    };
  };
  if (r) {
    const n = r + Te, a = o;
    o = (s) => s.startsWith(n) ? a(s.substring(n.length)) : {
      isExternal: !0,
      modifiers: [],
      hasImportantModifier: !1,
      baseClassName: s,
      maybePostfixModifierPosition: void 0
    };
  }
  if (t) {
    const n = o;
    o = (a) => t({
      className: a,
      parseClassName: n
    });
  }
  return o;
}, wr = (e) => e.endsWith(Ce) ? e.substring(0, e.length - 1) : e.startsWith(Ce) ? e.substring(1) : e, yr = (e) => {
  const r = Object.fromEntries(e.orderSensitiveModifiers.map((o) => [o, !0]));
  return (o) => {
    if (o.length <= 1)
      return o;
    const n = [];
    let a = [];
    return o.forEach((s) => {
      s[0] === "[" || r[s] ? (n.push(...a.sort(), s), a = []) : a.push(s);
    }), n.push(...a.sort()), n;
  };
}, kr = (e) => ({
  cache: br(e.cacheSize),
  parseClassName: vr(e),
  sortModifiers: yr(e),
  ...fr(e)
}), Er = /\s+/, Rr = (e, r) => {
  const {
    parseClassName: t,
    getClassGroupId: o,
    getConflictingClassGroupIds: n,
    sortModifiers: a
  } = r, s = [], c = e.trim().split(Er);
  let i = "";
  for (let m = c.length - 1; m >= 0; m -= 1) {
    const b = c[m], {
      isExternal: P,
      modifiers: w,
      hasImportantModifier: R,
      baseClassName: C,
      maybePostfixModifierPosition: S
    } = t(b);
    if (P) {
      i = b + (i.length > 0 ? " " + i : i);
      continue;
    }
    let j = !!S, H = o(j ? C.substring(0, S) : C);
    if (!H) {
      if (!j) {
        i = b + (i.length > 0 ? " " + i : i);
        continue;
      }
      if (H = o(C), !H) {
        i = b + (i.length > 0 ? " " + i : i);
        continue;
      }
      j = !1;
    }
    const re = a(w).join(":"), J = R ? re + Ce : re, D = J + H;
    if (s.includes(D))
      continue;
    s.push(D);
    const V = n(H, j);
    for (let I = 0; I < V.length; ++I) {
      const X = V[I];
      s.push(J + X);
    }
    i = b + (i.length > 0 ? " " + i : i);
  }
  return i;
};
function Sr() {
  let e = 0, r, t, o = "";
  for (; e < arguments.length; )
    (r = arguments[e++]) && (t = Xe(r)) && (o && (o += " "), o += t);
  return o;
}
const Xe = (e) => {
  if (typeof e == "string")
    return e;
  let r, t = "";
  for (let o = 0; o < e.length; o++)
    e[o] && (r = Xe(e[o])) && (t && (t += " "), t += r);
  return t;
};
function Cr(e, ...r) {
  let t, o, n, a = s;
  function s(i) {
    const m = r.reduce((b, P) => P(b), e());
    return t = kr(m), o = t.cache.get, n = t.cache.set, a = c, c(i);
  }
  function c(i) {
    const m = o(i);
    if (m)
      return m;
    const b = Rr(i, t);
    return n(i, b), b;
  }
  return function() {
    return a(Sr.apply(null, arguments));
  };
}
const M = (e) => {
  const r = (t) => t[e] || [];
  return r.isThemeGetter = !0, r;
}, Ke = /^\[(?:(\w[\w-]*):)?(.+)\]$/i, Ze = /^\((?:(\w[\w-]*):)?(.+)\)$/i, Tr = /^\d+\/\d+$/, Mr = /^(\d+(\.\d+)?)?(xs|sm|md|lg|xl)$/, Ar = /\d+(%|px|r?em|[sdl]?v([hwib]|min|max)|pt|pc|in|cm|mm|cap|ch|ex|r?lh|cq(w|h|i|b|min|max))|\b(calc|min|max|clamp)\(.+\)|^0$/, Pr = /^(rgba?|hsla?|hwb|(ok)?(lab|lch)|color-mix)\(.+\)$/, _r = /^(inset_)?-?((\d+)?\.?(\d+)[a-z]+|0)_-?((\d+)?\.?(\d+)[a-z]+|0)/, jr = /^(url|image|image-set|cross-fade|element|(repeating-)?(linear|radial|conic)-gradient)\(.+\)$/, ne = (e) => Tr.test(e), g = (e) => !!e && !Number.isNaN(Number(e)), U = (e) => !!e && Number.isInteger(Number(e)), Ee = (e) => e.endsWith("%") && g(e.slice(0, -1)), G = (e) => Mr.test(e), zr = () => !0, Nr = (e) => (
  // `colorFunctionRegex` check is necessary because color functions can have percentages in them which which would be incorrectly classified as lengths.
  // For example, `hsl(0 0% 0%)` would be classified as a length without this check.
  // I could also use lookbehind assertion in `lengthUnitRegex` but that isn't supported widely enough.
  Ar.test(e) && !Pr.test(e)
), Qe = () => !1, $r = (e) => _r.test(e), Or = (e) => jr.test(e), Hr = (e) => !d(e) && !u(e), Ir = (e) => se(e, tr, Qe), d = (e) => Ke.test(e), Q = (e) => se(e, or, Nr), Re = (e) => se(e, Vr, g), De = (e) => se(e, er, Qe), Lr = (e) => se(e, rr, Or), ge = (e) => se(e, nr, $r), u = (e) => Ze.test(e), ue = (e) => ae(e, or), Gr = (e) => ae(e, Yr), Ve = (e) => ae(e, er), Fr = (e) => ae(e, tr), Dr = (e) => ae(e, rr), be = (e) => ae(e, nr, !0), se = (e, r, t) => {
  const o = Ke.exec(e);
  return o ? o[1] ? r(o[1]) : t(o[2]) : !1;
}, ae = (e, r, t = !1) => {
  const o = Ze.exec(e);
  return o ? o[1] ? r(o[1]) : t : !1;
}, er = (e) => e === "position" || e === "percentage", rr = (e) => e === "image" || e === "url", tr = (e) => e === "length" || e === "size" || e === "bg-size", or = (e) => e === "length", Vr = (e) => e === "number", Yr = (e) => e === "family-name", nr = (e) => e === "shadow", Ur = () => {
  const e = M("color"), r = M("font"), t = M("text"), o = M("font-weight"), n = M("tracking"), a = M("leading"), s = M("breakpoint"), c = M("container"), i = M("spacing"), m = M("radius"), b = M("shadow"), P = M("inset-shadow"), w = M("text-shadow"), R = M("drop-shadow"), C = M("blur"), S = M("perspective"), j = M("aspect"), H = M("ease"), re = M("animate"), J = () => ["auto", "avoid", "all", "avoid-page", "page", "left", "right", "column"], D = () => [
    "center",
    "top",
    "bottom",
    "left",
    "right",
    "top-left",
    // Deprecated since Tailwind CSS v4.1.0, see https://github.com/tailwindlabs/tailwindcss/pull/17378
    "left-top",
    "top-right",
    // Deprecated since Tailwind CSS v4.1.0, see https://github.com/tailwindlabs/tailwindcss/pull/17378
    "right-top",
    "bottom-right",
    // Deprecated since Tailwind CSS v4.1.0, see https://github.com/tailwindlabs/tailwindcss/pull/17378
    "right-bottom",
    "bottom-left",
    // Deprecated since Tailwind CSS v4.1.0, see https://github.com/tailwindlabs/tailwindcss/pull/17378
    "left-bottom"
  ], V = () => [...D(), u, d], I = () => ["auto", "hidden", "clip", "visible", "scroll"], X = () => ["auto", "contain", "none"], p = () => [u, d, i], N = () => [ne, "full", "auto", ...p()], me = () => [U, "none", "subgrid", u, d], te = () => ["auto", {
    span: ["full", U, u, d]
  }, U, u, d], K = () => [U, "auto", u, d], fe = () => ["auto", "min", "max", "fr", u, d], Z = () => ["start", "end", "center", "between", "around", "evenly", "stretch", "baseline", "center-safe", "end-safe"], L = () => ["start", "end", "center", "stretch", "center-safe", "end-safe"], z = () => ["auto", ...p()], O = () => [ne, "auto", "full", "dvw", "dvh", "lvw", "lvh", "svw", "svh", "min", "max", "fit", ...p()], f = () => [e, u, d], le = () => [...D(), Ve, De, {
    position: [u, d]
  }], l = () => ["no-repeat", {
    repeat: ["", "x", "y", "space", "round"]
  }], h = () => ["auto", "cover", "contain", Fr, Ir, {
    size: [u, d]
  }], v = () => [Ee, ue, Q], x = () => [
    // Deprecated since Tailwind CSS v4.0.0
    "",
    "none",
    "full",
    m,
    u,
    d
  ], y = () => ["", g, ue, Q], _ = () => ["solid", "dashed", "dotted", "double"], oe = () => ["normal", "multiply", "screen", "overlay", "darken", "lighten", "color-dodge", "color-burn", "hard-light", "soft-light", "difference", "exclusion", "hue", "saturation", "color", "luminosity"], k = () => [g, Ee, Ve, De], T = () => [
    // Deprecated since Tailwind CSS v4.0.0
    "",
    "none",
    C,
    u,
    d
  ], $ = () => ["none", g, u, d], Y = () => ["none", g, u, d], ie = () => [g, u, d], pe = () => [ne, "full", ...p()];
  return {
    cacheSize: 500,
    theme: {
      animate: ["spin", "ping", "pulse", "bounce"],
      aspect: ["video"],
      blur: [G],
      breakpoint: [G],
      color: [zr],
      container: [G],
      "drop-shadow": [G],
      ease: ["in", "out", "in-out"],
      font: [Hr],
      "font-weight": ["thin", "extralight", "light", "normal", "medium", "semibold", "bold", "extrabold", "black"],
      "inset-shadow": [G],
      leading: ["none", "tight", "snug", "normal", "relaxed", "loose"],
      perspective: ["dramatic", "near", "normal", "midrange", "distant", "none"],
      radius: [G],
      shadow: [G],
      spacing: ["px", g],
      text: [G],
      "text-shadow": [G],
      tracking: ["tighter", "tight", "normal", "wide", "wider", "widest"]
    },
    classGroups: {
      // --------------
      // --- Layout ---
      // --------------
      /**
       * Aspect Ratio
       * @see https://tailwindcss.com/docs/aspect-ratio
       */
      aspect: [{
        aspect: ["auto", "square", ne, d, u, j]
      }],
      /**
       * Container
       * @see https://tailwindcss.com/docs/container
       * @deprecated since Tailwind CSS v4.0.0
       */
      container: ["container"],
      /**
       * Columns
       * @see https://tailwindcss.com/docs/columns
       */
      columns: [{
        columns: [g, d, u, c]
      }],
      /**
       * Break After
       * @see https://tailwindcss.com/docs/break-after
       */
      "break-after": [{
        "break-after": J()
      }],
      /**
       * Break Before
       * @see https://tailwindcss.com/docs/break-before
       */
      "break-before": [{
        "break-before": J()
      }],
      /**
       * Break Inside
       * @see https://tailwindcss.com/docs/break-inside
       */
      "break-inside": [{
        "break-inside": ["auto", "avoid", "avoid-page", "avoid-column"]
      }],
      /**
       * Box Decoration Break
       * @see https://tailwindcss.com/docs/box-decoration-break
       */
      "box-decoration": [{
        "box-decoration": ["slice", "clone"]
      }],
      /**
       * Box Sizing
       * @see https://tailwindcss.com/docs/box-sizing
       */
      box: [{
        box: ["border", "content"]
      }],
      /**
       * Display
       * @see https://tailwindcss.com/docs/display
       */
      display: ["block", "inline-block", "inline", "flex", "inline-flex", "table", "inline-table", "table-caption", "table-cell", "table-column", "table-column-group", "table-footer-group", "table-header-group", "table-row-group", "table-row", "flow-root", "grid", "inline-grid", "contents", "list-item", "hidden"],
      /**
       * Screen Reader Only
       * @see https://tailwindcss.com/docs/display#screen-reader-only
       */
      sr: ["sr-only", "not-sr-only"],
      /**
       * Floats
       * @see https://tailwindcss.com/docs/float
       */
      float: [{
        float: ["right", "left", "none", "start", "end"]
      }],
      /**
       * Clear
       * @see https://tailwindcss.com/docs/clear
       */
      clear: [{
        clear: ["left", "right", "both", "none", "start", "end"]
      }],
      /**
       * Isolation
       * @see https://tailwindcss.com/docs/isolation
       */
      isolation: ["isolate", "isolation-auto"],
      /**
       * Object Fit
       * @see https://tailwindcss.com/docs/object-fit
       */
      "object-fit": [{
        object: ["contain", "cover", "fill", "none", "scale-down"]
      }],
      /**
       * Object Position
       * @see https://tailwindcss.com/docs/object-position
       */
      "object-position": [{
        object: V()
      }],
      /**
       * Overflow
       * @see https://tailwindcss.com/docs/overflow
       */
      overflow: [{
        overflow: I()
      }],
      /**
       * Overflow X
       * @see https://tailwindcss.com/docs/overflow
       */
      "overflow-x": [{
        "overflow-x": I()
      }],
      /**
       * Overflow Y
       * @see https://tailwindcss.com/docs/overflow
       */
      "overflow-y": [{
        "overflow-y": I()
      }],
      /**
       * Overscroll Behavior
       * @see https://tailwindcss.com/docs/overscroll-behavior
       */
      overscroll: [{
        overscroll: X()
      }],
      /**
       * Overscroll Behavior X
       * @see https://tailwindcss.com/docs/overscroll-behavior
       */
      "overscroll-x": [{
        "overscroll-x": X()
      }],
      /**
       * Overscroll Behavior Y
       * @see https://tailwindcss.com/docs/overscroll-behavior
       */
      "overscroll-y": [{
        "overscroll-y": X()
      }],
      /**
       * Position
       * @see https://tailwindcss.com/docs/position
       */
      position: ["static", "fixed", "absolute", "relative", "sticky"],
      /**
       * Top / Right / Bottom / Left
       * @see https://tailwindcss.com/docs/top-right-bottom-left
       */
      inset: [{
        inset: N()
      }],
      /**
       * Right / Left
       * @see https://tailwindcss.com/docs/top-right-bottom-left
       */
      "inset-x": [{
        "inset-x": N()
      }],
      /**
       * Top / Bottom
       * @see https://tailwindcss.com/docs/top-right-bottom-left
       */
      "inset-y": [{
        "inset-y": N()
      }],
      /**
       * Start
       * @see https://tailwindcss.com/docs/top-right-bottom-left
       */
      start: [{
        start: N()
      }],
      /**
       * End
       * @see https://tailwindcss.com/docs/top-right-bottom-left
       */
      end: [{
        end: N()
      }],
      /**
       * Top
       * @see https://tailwindcss.com/docs/top-right-bottom-left
       */
      top: [{
        top: N()
      }],
      /**
       * Right
       * @see https://tailwindcss.com/docs/top-right-bottom-left
       */
      right: [{
        right: N()
      }],
      /**
       * Bottom
       * @see https://tailwindcss.com/docs/top-right-bottom-left
       */
      bottom: [{
        bottom: N()
      }],
      /**
       * Left
       * @see https://tailwindcss.com/docs/top-right-bottom-left
       */
      left: [{
        left: N()
      }],
      /**
       * Visibility
       * @see https://tailwindcss.com/docs/visibility
       */
      visibility: ["visible", "invisible", "collapse"],
      /**
       * Z-Index
       * @see https://tailwindcss.com/docs/z-index
       */
      z: [{
        z: [U, "auto", u, d]
      }],
      // ------------------------
      // --- Flexbox and Grid ---
      // ------------------------
      /**
       * Flex Basis
       * @see https://tailwindcss.com/docs/flex-basis
       */
      basis: [{
        basis: [ne, "full", "auto", c, ...p()]
      }],
      /**
       * Flex Direction
       * @see https://tailwindcss.com/docs/flex-direction
       */
      "flex-direction": [{
        flex: ["row", "row-reverse", "col", "col-reverse"]
      }],
      /**
       * Flex Wrap
       * @see https://tailwindcss.com/docs/flex-wrap
       */
      "flex-wrap": [{
        flex: ["nowrap", "wrap", "wrap-reverse"]
      }],
      /**
       * Flex
       * @see https://tailwindcss.com/docs/flex
       */
      flex: [{
        flex: [g, ne, "auto", "initial", "none", d]
      }],
      /**
       * Flex Grow
       * @see https://tailwindcss.com/docs/flex-grow
       */
      grow: [{
        grow: ["", g, u, d]
      }],
      /**
       * Flex Shrink
       * @see https://tailwindcss.com/docs/flex-shrink
       */
      shrink: [{
        shrink: ["", g, u, d]
      }],
      /**
       * Order
       * @see https://tailwindcss.com/docs/order
       */
      order: [{
        order: [U, "first", "last", "none", u, d]
      }],
      /**
       * Grid Template Columns
       * @see https://tailwindcss.com/docs/grid-template-columns
       */
      "grid-cols": [{
        "grid-cols": me()
      }],
      /**
       * Grid Column Start / End
       * @see https://tailwindcss.com/docs/grid-column
       */
      "col-start-end": [{
        col: te()
      }],
      /**
       * Grid Column Start
       * @see https://tailwindcss.com/docs/grid-column
       */
      "col-start": [{
        "col-start": K()
      }],
      /**
       * Grid Column End
       * @see https://tailwindcss.com/docs/grid-column
       */
      "col-end": [{
        "col-end": K()
      }],
      /**
       * Grid Template Rows
       * @see https://tailwindcss.com/docs/grid-template-rows
       */
      "grid-rows": [{
        "grid-rows": me()
      }],
      /**
       * Grid Row Start / End
       * @see https://tailwindcss.com/docs/grid-row
       */
      "row-start-end": [{
        row: te()
      }],
      /**
       * Grid Row Start
       * @see https://tailwindcss.com/docs/grid-row
       */
      "row-start": [{
        "row-start": K()
      }],
      /**
       * Grid Row End
       * @see https://tailwindcss.com/docs/grid-row
       */
      "row-end": [{
        "row-end": K()
      }],
      /**
       * Grid Auto Flow
       * @see https://tailwindcss.com/docs/grid-auto-flow
       */
      "grid-flow": [{
        "grid-flow": ["row", "col", "dense", "row-dense", "col-dense"]
      }],
      /**
       * Grid Auto Columns
       * @see https://tailwindcss.com/docs/grid-auto-columns
       */
      "auto-cols": [{
        "auto-cols": fe()
      }],
      /**
       * Grid Auto Rows
       * @see https://tailwindcss.com/docs/grid-auto-rows
       */
      "auto-rows": [{
        "auto-rows": fe()
      }],
      /**
       * Gap
       * @see https://tailwindcss.com/docs/gap
       */
      gap: [{
        gap: p()
      }],
      /**
       * Gap X
       * @see https://tailwindcss.com/docs/gap
       */
      "gap-x": [{
        "gap-x": p()
      }],
      /**
       * Gap Y
       * @see https://tailwindcss.com/docs/gap
       */
      "gap-y": [{
        "gap-y": p()
      }],
      /**
       * Justify Content
       * @see https://tailwindcss.com/docs/justify-content
       */
      "justify-content": [{
        justify: [...Z(), "normal"]
      }],
      /**
       * Justify Items
       * @see https://tailwindcss.com/docs/justify-items
       */
      "justify-items": [{
        "justify-items": [...L(), "normal"]
      }],
      /**
       * Justify Self
       * @see https://tailwindcss.com/docs/justify-self
       */
      "justify-self": [{
        "justify-self": ["auto", ...L()]
      }],
      /**
       * Align Content
       * @see https://tailwindcss.com/docs/align-content
       */
      "align-content": [{
        content: ["normal", ...Z()]
      }],
      /**
       * Align Items
       * @see https://tailwindcss.com/docs/align-items
       */
      "align-items": [{
        items: [...L(), {
          baseline: ["", "last"]
        }]
      }],
      /**
       * Align Self
       * @see https://tailwindcss.com/docs/align-self
       */
      "align-self": [{
        self: ["auto", ...L(), {
          baseline: ["", "last"]
        }]
      }],
      /**
       * Place Content
       * @see https://tailwindcss.com/docs/place-content
       */
      "place-content": [{
        "place-content": Z()
      }],
      /**
       * Place Items
       * @see https://tailwindcss.com/docs/place-items
       */
      "place-items": [{
        "place-items": [...L(), "baseline"]
      }],
      /**
       * Place Self
       * @see https://tailwindcss.com/docs/place-self
       */
      "place-self": [{
        "place-self": ["auto", ...L()]
      }],
      // Spacing
      /**
       * Padding
       * @see https://tailwindcss.com/docs/padding
       */
      p: [{
        p: p()
      }],
      /**
       * Padding X
       * @see https://tailwindcss.com/docs/padding
       */
      px: [{
        px: p()
      }],
      /**
       * Padding Y
       * @see https://tailwindcss.com/docs/padding
       */
      py: [{
        py: p()
      }],
      /**
       * Padding Start
       * @see https://tailwindcss.com/docs/padding
       */
      ps: [{
        ps: p()
      }],
      /**
       * Padding End
       * @see https://tailwindcss.com/docs/padding
       */
      pe: [{
        pe: p()
      }],
      /**
       * Padding Top
       * @see https://tailwindcss.com/docs/padding
       */
      pt: [{
        pt: p()
      }],
      /**
       * Padding Right
       * @see https://tailwindcss.com/docs/padding
       */
      pr: [{
        pr: p()
      }],
      /**
       * Padding Bottom
       * @see https://tailwindcss.com/docs/padding
       */
      pb: [{
        pb: p()
      }],
      /**
       * Padding Left
       * @see https://tailwindcss.com/docs/padding
       */
      pl: [{
        pl: p()
      }],
      /**
       * Margin
       * @see https://tailwindcss.com/docs/margin
       */
      m: [{
        m: z()
      }],
      /**
       * Margin X
       * @see https://tailwindcss.com/docs/margin
       */
      mx: [{
        mx: z()
      }],
      /**
       * Margin Y
       * @see https://tailwindcss.com/docs/margin
       */
      my: [{
        my: z()
      }],
      /**
       * Margin Start
       * @see https://tailwindcss.com/docs/margin
       */
      ms: [{
        ms: z()
      }],
      /**
       * Margin End
       * @see https://tailwindcss.com/docs/margin
       */
      me: [{
        me: z()
      }],
      /**
       * Margin Top
       * @see https://tailwindcss.com/docs/margin
       */
      mt: [{
        mt: z()
      }],
      /**
       * Margin Right
       * @see https://tailwindcss.com/docs/margin
       */
      mr: [{
        mr: z()
      }],
      /**
       * Margin Bottom
       * @see https://tailwindcss.com/docs/margin
       */
      mb: [{
        mb: z()
      }],
      /**
       * Margin Left
       * @see https://tailwindcss.com/docs/margin
       */
      ml: [{
        ml: z()
      }],
      /**
       * Space Between X
       * @see https://tailwindcss.com/docs/margin#adding-space-between-children
       */
      "space-x": [{
        "space-x": p()
      }],
      /**
       * Space Between X Reverse
       * @see https://tailwindcss.com/docs/margin#adding-space-between-children
       */
      "space-x-reverse": ["space-x-reverse"],
      /**
       * Space Between Y
       * @see https://tailwindcss.com/docs/margin#adding-space-between-children
       */
      "space-y": [{
        "space-y": p()
      }],
      /**
       * Space Between Y Reverse
       * @see https://tailwindcss.com/docs/margin#adding-space-between-children
       */
      "space-y-reverse": ["space-y-reverse"],
      // --------------
      // --- Sizing ---
      // --------------
      /**
       * Size
       * @see https://tailwindcss.com/docs/width#setting-both-width-and-height
       */
      size: [{
        size: O()
      }],
      /**
       * Width
       * @see https://tailwindcss.com/docs/width
       */
      w: [{
        w: [c, "screen", ...O()]
      }],
      /**
       * Min-Width
       * @see https://tailwindcss.com/docs/min-width
       */
      "min-w": [{
        "min-w": [
          c,
          "screen",
          /** Deprecated. @see https://github.com/tailwindlabs/tailwindcss.com/issues/2027#issuecomment-2620152757 */
          "none",
          ...O()
        ]
      }],
      /**
       * Max-Width
       * @see https://tailwindcss.com/docs/max-width
       */
      "max-w": [{
        "max-w": [
          c,
          "screen",
          "none",
          /** Deprecated since Tailwind CSS v4.0.0. @see https://github.com/tailwindlabs/tailwindcss.com/issues/2027#issuecomment-2620152757 */
          "prose",
          /** Deprecated since Tailwind CSS v4.0.0. @see https://github.com/tailwindlabs/tailwindcss.com/issues/2027#issuecomment-2620152757 */
          {
            screen: [s]
          },
          ...O()
        ]
      }],
      /**
       * Height
       * @see https://tailwindcss.com/docs/height
       */
      h: [{
        h: ["screen", "lh", ...O()]
      }],
      /**
       * Min-Height
       * @see https://tailwindcss.com/docs/min-height
       */
      "min-h": [{
        "min-h": ["screen", "lh", "none", ...O()]
      }],
      /**
       * Max-Height
       * @see https://tailwindcss.com/docs/max-height
       */
      "max-h": [{
        "max-h": ["screen", "lh", ...O()]
      }],
      // ------------------
      // --- Typography ---
      // ------------------
      /**
       * Font Size
       * @see https://tailwindcss.com/docs/font-size
       */
      "font-size": [{
        text: ["base", t, ue, Q]
      }],
      /**
       * Font Smoothing
       * @see https://tailwindcss.com/docs/font-smoothing
       */
      "font-smoothing": ["antialiased", "subpixel-antialiased"],
      /**
       * Font Style
       * @see https://tailwindcss.com/docs/font-style
       */
      "font-style": ["italic", "not-italic"],
      /**
       * Font Weight
       * @see https://tailwindcss.com/docs/font-weight
       */
      "font-weight": [{
        font: [o, u, Re]
      }],
      /**
       * Font Stretch
       * @see https://tailwindcss.com/docs/font-stretch
       */
      "font-stretch": [{
        "font-stretch": ["ultra-condensed", "extra-condensed", "condensed", "semi-condensed", "normal", "semi-expanded", "expanded", "extra-expanded", "ultra-expanded", Ee, d]
      }],
      /**
       * Font Family
       * @see https://tailwindcss.com/docs/font-family
       */
      "font-family": [{
        font: [Gr, d, r]
      }],
      /**
       * Font Variant Numeric
       * @see https://tailwindcss.com/docs/font-variant-numeric
       */
      "fvn-normal": ["normal-nums"],
      /**
       * Font Variant Numeric
       * @see https://tailwindcss.com/docs/font-variant-numeric
       */
      "fvn-ordinal": ["ordinal"],
      /**
       * Font Variant Numeric
       * @see https://tailwindcss.com/docs/font-variant-numeric
       */
      "fvn-slashed-zero": ["slashed-zero"],
      /**
       * Font Variant Numeric
       * @see https://tailwindcss.com/docs/font-variant-numeric
       */
      "fvn-figure": ["lining-nums", "oldstyle-nums"],
      /**
       * Font Variant Numeric
       * @see https://tailwindcss.com/docs/font-variant-numeric
       */
      "fvn-spacing": ["proportional-nums", "tabular-nums"],
      /**
       * Font Variant Numeric
       * @see https://tailwindcss.com/docs/font-variant-numeric
       */
      "fvn-fraction": ["diagonal-fractions", "stacked-fractions"],
      /**
       * Letter Spacing
       * @see https://tailwindcss.com/docs/letter-spacing
       */
      tracking: [{
        tracking: [n, u, d]
      }],
      /**
       * Line Clamp
       * @see https://tailwindcss.com/docs/line-clamp
       */
      "line-clamp": [{
        "line-clamp": [g, "none", u, Re]
      }],
      /**
       * Line Height
       * @see https://tailwindcss.com/docs/line-height
       */
      leading: [{
        leading: [
          /** Deprecated since Tailwind CSS v4.0.0. @see https://github.com/tailwindlabs/tailwindcss.com/issues/2027#issuecomment-2620152757 */
          a,
          ...p()
        ]
      }],
      /**
       * List Style Image
       * @see https://tailwindcss.com/docs/list-style-image
       */
      "list-image": [{
        "list-image": ["none", u, d]
      }],
      /**
       * List Style Position
       * @see https://tailwindcss.com/docs/list-style-position
       */
      "list-style-position": [{
        list: ["inside", "outside"]
      }],
      /**
       * List Style Type
       * @see https://tailwindcss.com/docs/list-style-type
       */
      "list-style-type": [{
        list: ["disc", "decimal", "none", u, d]
      }],
      /**
       * Text Alignment
       * @see https://tailwindcss.com/docs/text-align
       */
      "text-alignment": [{
        text: ["left", "center", "right", "justify", "start", "end"]
      }],
      /**
       * Placeholder Color
       * @deprecated since Tailwind CSS v3.0.0
       * @see https://v3.tailwindcss.com/docs/placeholder-color
       */
      "placeholder-color": [{
        placeholder: f()
      }],
      /**
       * Text Color
       * @see https://tailwindcss.com/docs/text-color
       */
      "text-color": [{
        text: f()
      }],
      /**
       * Text Decoration
       * @see https://tailwindcss.com/docs/text-decoration
       */
      "text-decoration": ["underline", "overline", "line-through", "no-underline"],
      /**
       * Text Decoration Style
       * @see https://tailwindcss.com/docs/text-decoration-style
       */
      "text-decoration-style": [{
        decoration: [..._(), "wavy"]
      }],
      /**
       * Text Decoration Thickness
       * @see https://tailwindcss.com/docs/text-decoration-thickness
       */
      "text-decoration-thickness": [{
        decoration: [g, "from-font", "auto", u, Q]
      }],
      /**
       * Text Decoration Color
       * @see https://tailwindcss.com/docs/text-decoration-color
       */
      "text-decoration-color": [{
        decoration: f()
      }],
      /**
       * Text Underline Offset
       * @see https://tailwindcss.com/docs/text-underline-offset
       */
      "underline-offset": [{
        "underline-offset": [g, "auto", u, d]
      }],
      /**
       * Text Transform
       * @see https://tailwindcss.com/docs/text-transform
       */
      "text-transform": ["uppercase", "lowercase", "capitalize", "normal-case"],
      /**
       * Text Overflow
       * @see https://tailwindcss.com/docs/text-overflow
       */
      "text-overflow": ["truncate", "text-ellipsis", "text-clip"],
      /**
       * Text Wrap
       * @see https://tailwindcss.com/docs/text-wrap
       */
      "text-wrap": [{
        text: ["wrap", "nowrap", "balance", "pretty"]
      }],
      /**
       * Text Indent
       * @see https://tailwindcss.com/docs/text-indent
       */
      indent: [{
        indent: p()
      }],
      /**
       * Vertical Alignment
       * @see https://tailwindcss.com/docs/vertical-align
       */
      "vertical-align": [{
        align: ["baseline", "top", "middle", "bottom", "text-top", "text-bottom", "sub", "super", u, d]
      }],
      /**
       * Whitespace
       * @see https://tailwindcss.com/docs/whitespace
       */
      whitespace: [{
        whitespace: ["normal", "nowrap", "pre", "pre-line", "pre-wrap", "break-spaces"]
      }],
      /**
       * Word Break
       * @see https://tailwindcss.com/docs/word-break
       */
      break: [{
        break: ["normal", "words", "all", "keep"]
      }],
      /**
       * Overflow Wrap
       * @see https://tailwindcss.com/docs/overflow-wrap
       */
      wrap: [{
        wrap: ["break-word", "anywhere", "normal"]
      }],
      /**
       * Hyphens
       * @see https://tailwindcss.com/docs/hyphens
       */
      hyphens: [{
        hyphens: ["none", "manual", "auto"]
      }],
      /**
       * Content
       * @see https://tailwindcss.com/docs/content
       */
      content: [{
        content: ["none", u, d]
      }],
      // -------------------
      // --- Backgrounds ---
      // -------------------
      /**
       * Background Attachment
       * @see https://tailwindcss.com/docs/background-attachment
       */
      "bg-attachment": [{
        bg: ["fixed", "local", "scroll"]
      }],
      /**
       * Background Clip
       * @see https://tailwindcss.com/docs/background-clip
       */
      "bg-clip": [{
        "bg-clip": ["border", "padding", "content", "text"]
      }],
      /**
       * Background Origin
       * @see https://tailwindcss.com/docs/background-origin
       */
      "bg-origin": [{
        "bg-origin": ["border", "padding", "content"]
      }],
      /**
       * Background Position
       * @see https://tailwindcss.com/docs/background-position
       */
      "bg-position": [{
        bg: le()
      }],
      /**
       * Background Repeat
       * @see https://tailwindcss.com/docs/background-repeat
       */
      "bg-repeat": [{
        bg: l()
      }],
      /**
       * Background Size
       * @see https://tailwindcss.com/docs/background-size
       */
      "bg-size": [{
        bg: h()
      }],
      /**
       * Background Image
       * @see https://tailwindcss.com/docs/background-image
       */
      "bg-image": [{
        bg: ["none", {
          linear: [{
            to: ["t", "tr", "r", "br", "b", "bl", "l", "tl"]
          }, U, u, d],
          radial: ["", u, d],
          conic: [U, u, d]
        }, Dr, Lr]
      }],
      /**
       * Background Color
       * @see https://tailwindcss.com/docs/background-color
       */
      "bg-color": [{
        bg: f()
      }],
      /**
       * Gradient Color Stops From Position
       * @see https://tailwindcss.com/docs/gradient-color-stops
       */
      "gradient-from-pos": [{
        from: v()
      }],
      /**
       * Gradient Color Stops Via Position
       * @see https://tailwindcss.com/docs/gradient-color-stops
       */
      "gradient-via-pos": [{
        via: v()
      }],
      /**
       * Gradient Color Stops To Position
       * @see https://tailwindcss.com/docs/gradient-color-stops
       */
      "gradient-to-pos": [{
        to: v()
      }],
      /**
       * Gradient Color Stops From
       * @see https://tailwindcss.com/docs/gradient-color-stops
       */
      "gradient-from": [{
        from: f()
      }],
      /**
       * Gradient Color Stops Via
       * @see https://tailwindcss.com/docs/gradient-color-stops
       */
      "gradient-via": [{
        via: f()
      }],
      /**
       * Gradient Color Stops To
       * @see https://tailwindcss.com/docs/gradient-color-stops
       */
      "gradient-to": [{
        to: f()
      }],
      // ---------------
      // --- Borders ---
      // ---------------
      /**
       * Border Radius
       * @see https://tailwindcss.com/docs/border-radius
       */
      rounded: [{
        rounded: x()
      }],
      /**
       * Border Radius Start
       * @see https://tailwindcss.com/docs/border-radius
       */
      "rounded-s": [{
        "rounded-s": x()
      }],
      /**
       * Border Radius End
       * @see https://tailwindcss.com/docs/border-radius
       */
      "rounded-e": [{
        "rounded-e": x()
      }],
      /**
       * Border Radius Top
       * @see https://tailwindcss.com/docs/border-radius
       */
      "rounded-t": [{
        "rounded-t": x()
      }],
      /**
       * Border Radius Right
       * @see https://tailwindcss.com/docs/border-radius
       */
      "rounded-r": [{
        "rounded-r": x()
      }],
      /**
       * Border Radius Bottom
       * @see https://tailwindcss.com/docs/border-radius
       */
      "rounded-b": [{
        "rounded-b": x()
      }],
      /**
       * Border Radius Left
       * @see https://tailwindcss.com/docs/border-radius
       */
      "rounded-l": [{
        "rounded-l": x()
      }],
      /**
       * Border Radius Start Start
       * @see https://tailwindcss.com/docs/border-radius
       */
      "rounded-ss": [{
        "rounded-ss": x()
      }],
      /**
       * Border Radius Start End
       * @see https://tailwindcss.com/docs/border-radius
       */
      "rounded-se": [{
        "rounded-se": x()
      }],
      /**
       * Border Radius End End
       * @see https://tailwindcss.com/docs/border-radius
       */
      "rounded-ee": [{
        "rounded-ee": x()
      }],
      /**
       * Border Radius End Start
       * @see https://tailwindcss.com/docs/border-radius
       */
      "rounded-es": [{
        "rounded-es": x()
      }],
      /**
       * Border Radius Top Left
       * @see https://tailwindcss.com/docs/border-radius
       */
      "rounded-tl": [{
        "rounded-tl": x()
      }],
      /**
       * Border Radius Top Right
       * @see https://tailwindcss.com/docs/border-radius
       */
      "rounded-tr": [{
        "rounded-tr": x()
      }],
      /**
       * Border Radius Bottom Right
       * @see https://tailwindcss.com/docs/border-radius
       */
      "rounded-br": [{
        "rounded-br": x()
      }],
      /**
       * Border Radius Bottom Left
       * @see https://tailwindcss.com/docs/border-radius
       */
      "rounded-bl": [{
        "rounded-bl": x()
      }],
      /**
       * Border Width
       * @see https://tailwindcss.com/docs/border-width
       */
      "border-w": [{
        border: y()
      }],
      /**
       * Border Width X
       * @see https://tailwindcss.com/docs/border-width
       */
      "border-w-x": [{
        "border-x": y()
      }],
      /**
       * Border Width Y
       * @see https://tailwindcss.com/docs/border-width
       */
      "border-w-y": [{
        "border-y": y()
      }],
      /**
       * Border Width Start
       * @see https://tailwindcss.com/docs/border-width
       */
      "border-w-s": [{
        "border-s": y()
      }],
      /**
       * Border Width End
       * @see https://tailwindcss.com/docs/border-width
       */
      "border-w-e": [{
        "border-e": y()
      }],
      /**
       * Border Width Top
       * @see https://tailwindcss.com/docs/border-width
       */
      "border-w-t": [{
        "border-t": y()
      }],
      /**
       * Border Width Right
       * @see https://tailwindcss.com/docs/border-width
       */
      "border-w-r": [{
        "border-r": y()
      }],
      /**
       * Border Width Bottom
       * @see https://tailwindcss.com/docs/border-width
       */
      "border-w-b": [{
        "border-b": y()
      }],
      /**
       * Border Width Left
       * @see https://tailwindcss.com/docs/border-width
       */
      "border-w-l": [{
        "border-l": y()
      }],
      /**
       * Divide Width X
       * @see https://tailwindcss.com/docs/border-width#between-children
       */
      "divide-x": [{
        "divide-x": y()
      }],
      /**
       * Divide Width X Reverse
       * @see https://tailwindcss.com/docs/border-width#between-children
       */
      "divide-x-reverse": ["divide-x-reverse"],
      /**
       * Divide Width Y
       * @see https://tailwindcss.com/docs/border-width#between-children
       */
      "divide-y": [{
        "divide-y": y()
      }],
      /**
       * Divide Width Y Reverse
       * @see https://tailwindcss.com/docs/border-width#between-children
       */
      "divide-y-reverse": ["divide-y-reverse"],
      /**
       * Border Style
       * @see https://tailwindcss.com/docs/border-style
       */
      "border-style": [{
        border: [..._(), "hidden", "none"]
      }],
      /**
       * Divide Style
       * @see https://tailwindcss.com/docs/border-style#setting-the-divider-style
       */
      "divide-style": [{
        divide: [..._(), "hidden", "none"]
      }],
      /**
       * Border Color
       * @see https://tailwindcss.com/docs/border-color
       */
      "border-color": [{
        border: f()
      }],
      /**
       * Border Color X
       * @see https://tailwindcss.com/docs/border-color
       */
      "border-color-x": [{
        "border-x": f()
      }],
      /**
       * Border Color Y
       * @see https://tailwindcss.com/docs/border-color
       */
      "border-color-y": [{
        "border-y": f()
      }],
      /**
       * Border Color S
       * @see https://tailwindcss.com/docs/border-color
       */
      "border-color-s": [{
        "border-s": f()
      }],
      /**
       * Border Color E
       * @see https://tailwindcss.com/docs/border-color
       */
      "border-color-e": [{
        "border-e": f()
      }],
      /**
       * Border Color Top
       * @see https://tailwindcss.com/docs/border-color
       */
      "border-color-t": [{
        "border-t": f()
      }],
      /**
       * Border Color Right
       * @see https://tailwindcss.com/docs/border-color
       */
      "border-color-r": [{
        "border-r": f()
      }],
      /**
       * Border Color Bottom
       * @see https://tailwindcss.com/docs/border-color
       */
      "border-color-b": [{
        "border-b": f()
      }],
      /**
       * Border Color Left
       * @see https://tailwindcss.com/docs/border-color
       */
      "border-color-l": [{
        "border-l": f()
      }],
      /**
       * Divide Color
       * @see https://tailwindcss.com/docs/divide-color
       */
      "divide-color": [{
        divide: f()
      }],
      /**
       * Outline Style
       * @see https://tailwindcss.com/docs/outline-style
       */
      "outline-style": [{
        outline: [..._(), "none", "hidden"]
      }],
      /**
       * Outline Offset
       * @see https://tailwindcss.com/docs/outline-offset
       */
      "outline-offset": [{
        "outline-offset": [g, u, d]
      }],
      /**
       * Outline Width
       * @see https://tailwindcss.com/docs/outline-width
       */
      "outline-w": [{
        outline: ["", g, ue, Q]
      }],
      /**
       * Outline Color
       * @see https://tailwindcss.com/docs/outline-color
       */
      "outline-color": [{
        outline: f()
      }],
      // ---------------
      // --- Effects ---
      // ---------------
      /**
       * Box Shadow
       * @see https://tailwindcss.com/docs/box-shadow
       */
      shadow: [{
        shadow: [
          // Deprecated since Tailwind CSS v4.0.0
          "",
          "none",
          b,
          be,
          ge
        ]
      }],
      /**
       * Box Shadow Color
       * @see https://tailwindcss.com/docs/box-shadow#setting-the-shadow-color
       */
      "shadow-color": [{
        shadow: f()
      }],
      /**
       * Inset Box Shadow
       * @see https://tailwindcss.com/docs/box-shadow#adding-an-inset-shadow
       */
      "inset-shadow": [{
        "inset-shadow": ["none", P, be, ge]
      }],
      /**
       * Inset Box Shadow Color
       * @see https://tailwindcss.com/docs/box-shadow#setting-the-inset-shadow-color
       */
      "inset-shadow-color": [{
        "inset-shadow": f()
      }],
      /**
       * Ring Width
       * @see https://tailwindcss.com/docs/box-shadow#adding-a-ring
       */
      "ring-w": [{
        ring: y()
      }],
      /**
       * Ring Width Inset
       * @see https://v3.tailwindcss.com/docs/ring-width#inset-rings
       * @deprecated since Tailwind CSS v4.0.0
       * @see https://github.com/tailwindlabs/tailwindcss/blob/v4.0.0/packages/tailwindcss/src/utilities.ts#L4158
       */
      "ring-w-inset": ["ring-inset"],
      /**
       * Ring Color
       * @see https://tailwindcss.com/docs/box-shadow#setting-the-ring-color
       */
      "ring-color": [{
        ring: f()
      }],
      /**
       * Ring Offset Width
       * @see https://v3.tailwindcss.com/docs/ring-offset-width
       * @deprecated since Tailwind CSS v4.0.0
       * @see https://github.com/tailwindlabs/tailwindcss/blob/v4.0.0/packages/tailwindcss/src/utilities.ts#L4158
       */
      "ring-offset-w": [{
        "ring-offset": [g, Q]
      }],
      /**
       * Ring Offset Color
       * @see https://v3.tailwindcss.com/docs/ring-offset-color
       * @deprecated since Tailwind CSS v4.0.0
       * @see https://github.com/tailwindlabs/tailwindcss/blob/v4.0.0/packages/tailwindcss/src/utilities.ts#L4158
       */
      "ring-offset-color": [{
        "ring-offset": f()
      }],
      /**
       * Inset Ring Width
       * @see https://tailwindcss.com/docs/box-shadow#adding-an-inset-ring
       */
      "inset-ring-w": [{
        "inset-ring": y()
      }],
      /**
       * Inset Ring Color
       * @see https://tailwindcss.com/docs/box-shadow#setting-the-inset-ring-color
       */
      "inset-ring-color": [{
        "inset-ring": f()
      }],
      /**
       * Text Shadow
       * @see https://tailwindcss.com/docs/text-shadow
       */
      "text-shadow": [{
        "text-shadow": ["none", w, be, ge]
      }],
      /**
       * Text Shadow Color
       * @see https://tailwindcss.com/docs/text-shadow#setting-the-shadow-color
       */
      "text-shadow-color": [{
        "text-shadow": f()
      }],
      /**
       * Opacity
       * @see https://tailwindcss.com/docs/opacity
       */
      opacity: [{
        opacity: [g, u, d]
      }],
      /**
       * Mix Blend Mode
       * @see https://tailwindcss.com/docs/mix-blend-mode
       */
      "mix-blend": [{
        "mix-blend": [...oe(), "plus-darker", "plus-lighter"]
      }],
      /**
       * Background Blend Mode
       * @see https://tailwindcss.com/docs/background-blend-mode
       */
      "bg-blend": [{
        "bg-blend": oe()
      }],
      /**
       * Mask Clip
       * @see https://tailwindcss.com/docs/mask-clip
       */
      "mask-clip": [{
        "mask-clip": ["border", "padding", "content", "fill", "stroke", "view"]
      }, "mask-no-clip"],
      /**
       * Mask Composite
       * @see https://tailwindcss.com/docs/mask-composite
       */
      "mask-composite": [{
        mask: ["add", "subtract", "intersect", "exclude"]
      }],
      /**
       * Mask Image
       * @see https://tailwindcss.com/docs/mask-image
       */
      "mask-image-linear-pos": [{
        "mask-linear": [g]
      }],
      "mask-image-linear-from-pos": [{
        "mask-linear-from": k()
      }],
      "mask-image-linear-to-pos": [{
        "mask-linear-to": k()
      }],
      "mask-image-linear-from-color": [{
        "mask-linear-from": f()
      }],
      "mask-image-linear-to-color": [{
        "mask-linear-to": f()
      }],
      "mask-image-t-from-pos": [{
        "mask-t-from": k()
      }],
      "mask-image-t-to-pos": [{
        "mask-t-to": k()
      }],
      "mask-image-t-from-color": [{
        "mask-t-from": f()
      }],
      "mask-image-t-to-color": [{
        "mask-t-to": f()
      }],
      "mask-image-r-from-pos": [{
        "mask-r-from": k()
      }],
      "mask-image-r-to-pos": [{
        "mask-r-to": k()
      }],
      "mask-image-r-from-color": [{
        "mask-r-from": f()
      }],
      "mask-image-r-to-color": [{
        "mask-r-to": f()
      }],
      "mask-image-b-from-pos": [{
        "mask-b-from": k()
      }],
      "mask-image-b-to-pos": [{
        "mask-b-to": k()
      }],
      "mask-image-b-from-color": [{
        "mask-b-from": f()
      }],
      "mask-image-b-to-color": [{
        "mask-b-to": f()
      }],
      "mask-image-l-from-pos": [{
        "mask-l-from": k()
      }],
      "mask-image-l-to-pos": [{
        "mask-l-to": k()
      }],
      "mask-image-l-from-color": [{
        "mask-l-from": f()
      }],
      "mask-image-l-to-color": [{
        "mask-l-to": f()
      }],
      "mask-image-x-from-pos": [{
        "mask-x-from": k()
      }],
      "mask-image-x-to-pos": [{
        "mask-x-to": k()
      }],
      "mask-image-x-from-color": [{
        "mask-x-from": f()
      }],
      "mask-image-x-to-color": [{
        "mask-x-to": f()
      }],
      "mask-image-y-from-pos": [{
        "mask-y-from": k()
      }],
      "mask-image-y-to-pos": [{
        "mask-y-to": k()
      }],
      "mask-image-y-from-color": [{
        "mask-y-from": f()
      }],
      "mask-image-y-to-color": [{
        "mask-y-to": f()
      }],
      "mask-image-radial": [{
        "mask-radial": [u, d]
      }],
      "mask-image-radial-from-pos": [{
        "mask-radial-from": k()
      }],
      "mask-image-radial-to-pos": [{
        "mask-radial-to": k()
      }],
      "mask-image-radial-from-color": [{
        "mask-radial-from": f()
      }],
      "mask-image-radial-to-color": [{
        "mask-radial-to": f()
      }],
      "mask-image-radial-shape": [{
        "mask-radial": ["circle", "ellipse"]
      }],
      "mask-image-radial-size": [{
        "mask-radial": [{
          closest: ["side", "corner"],
          farthest: ["side", "corner"]
        }]
      }],
      "mask-image-radial-pos": [{
        "mask-radial-at": D()
      }],
      "mask-image-conic-pos": [{
        "mask-conic": [g]
      }],
      "mask-image-conic-from-pos": [{
        "mask-conic-from": k()
      }],
      "mask-image-conic-to-pos": [{
        "mask-conic-to": k()
      }],
      "mask-image-conic-from-color": [{
        "mask-conic-from": f()
      }],
      "mask-image-conic-to-color": [{
        "mask-conic-to": f()
      }],
      /**
       * Mask Mode
       * @see https://tailwindcss.com/docs/mask-mode
       */
      "mask-mode": [{
        mask: ["alpha", "luminance", "match"]
      }],
      /**
       * Mask Origin
       * @see https://tailwindcss.com/docs/mask-origin
       */
      "mask-origin": [{
        "mask-origin": ["border", "padding", "content", "fill", "stroke", "view"]
      }],
      /**
       * Mask Position
       * @see https://tailwindcss.com/docs/mask-position
       */
      "mask-position": [{
        mask: le()
      }],
      /**
       * Mask Repeat
       * @see https://tailwindcss.com/docs/mask-repeat
       */
      "mask-repeat": [{
        mask: l()
      }],
      /**
       * Mask Size
       * @see https://tailwindcss.com/docs/mask-size
       */
      "mask-size": [{
        mask: h()
      }],
      /**
       * Mask Type
       * @see https://tailwindcss.com/docs/mask-type
       */
      "mask-type": [{
        "mask-type": ["alpha", "luminance"]
      }],
      /**
       * Mask Image
       * @see https://tailwindcss.com/docs/mask-image
       */
      "mask-image": [{
        mask: ["none", u, d]
      }],
      // ---------------
      // --- Filters ---
      // ---------------
      /**
       * Filter
       * @see https://tailwindcss.com/docs/filter
       */
      filter: [{
        filter: [
          // Deprecated since Tailwind CSS v3.0.0
          "",
          "none",
          u,
          d
        ]
      }],
      /**
       * Blur
       * @see https://tailwindcss.com/docs/blur
       */
      blur: [{
        blur: T()
      }],
      /**
       * Brightness
       * @see https://tailwindcss.com/docs/brightness
       */
      brightness: [{
        brightness: [g, u, d]
      }],
      /**
       * Contrast
       * @see https://tailwindcss.com/docs/contrast
       */
      contrast: [{
        contrast: [g, u, d]
      }],
      /**
       * Drop Shadow
       * @see https://tailwindcss.com/docs/drop-shadow
       */
      "drop-shadow": [{
        "drop-shadow": [
          // Deprecated since Tailwind CSS v4.0.0
          "",
          "none",
          R,
          be,
          ge
        ]
      }],
      /**
       * Drop Shadow Color
       * @see https://tailwindcss.com/docs/filter-drop-shadow#setting-the-shadow-color
       */
      "drop-shadow-color": [{
        "drop-shadow": f()
      }],
      /**
       * Grayscale
       * @see https://tailwindcss.com/docs/grayscale
       */
      grayscale: [{
        grayscale: ["", g, u, d]
      }],
      /**
       * Hue Rotate
       * @see https://tailwindcss.com/docs/hue-rotate
       */
      "hue-rotate": [{
        "hue-rotate": [g, u, d]
      }],
      /**
       * Invert
       * @see https://tailwindcss.com/docs/invert
       */
      invert: [{
        invert: ["", g, u, d]
      }],
      /**
       * Saturate
       * @see https://tailwindcss.com/docs/saturate
       */
      saturate: [{
        saturate: [g, u, d]
      }],
      /**
       * Sepia
       * @see https://tailwindcss.com/docs/sepia
       */
      sepia: [{
        sepia: ["", g, u, d]
      }],
      /**
       * Backdrop Filter
       * @see https://tailwindcss.com/docs/backdrop-filter
       */
      "backdrop-filter": [{
        "backdrop-filter": [
          // Deprecated since Tailwind CSS v3.0.0
          "",
          "none",
          u,
          d
        ]
      }],
      /**
       * Backdrop Blur
       * @see https://tailwindcss.com/docs/backdrop-blur
       */
      "backdrop-blur": [{
        "backdrop-blur": T()
      }],
      /**
       * Backdrop Brightness
       * @see https://tailwindcss.com/docs/backdrop-brightness
       */
      "backdrop-brightness": [{
        "backdrop-brightness": [g, u, d]
      }],
      /**
       * Backdrop Contrast
       * @see https://tailwindcss.com/docs/backdrop-contrast
       */
      "backdrop-contrast": [{
        "backdrop-contrast": [g, u, d]
      }],
      /**
       * Backdrop Grayscale
       * @see https://tailwindcss.com/docs/backdrop-grayscale
       */
      "backdrop-grayscale": [{
        "backdrop-grayscale": ["", g, u, d]
      }],
      /**
       * Backdrop Hue Rotate
       * @see https://tailwindcss.com/docs/backdrop-hue-rotate
       */
      "backdrop-hue-rotate": [{
        "backdrop-hue-rotate": [g, u, d]
      }],
      /**
       * Backdrop Invert
       * @see https://tailwindcss.com/docs/backdrop-invert
       */
      "backdrop-invert": [{
        "backdrop-invert": ["", g, u, d]
      }],
      /**
       * Backdrop Opacity
       * @see https://tailwindcss.com/docs/backdrop-opacity
       */
      "backdrop-opacity": [{
        "backdrop-opacity": [g, u, d]
      }],
      /**
       * Backdrop Saturate
       * @see https://tailwindcss.com/docs/backdrop-saturate
       */
      "backdrop-saturate": [{
        "backdrop-saturate": [g, u, d]
      }],
      /**
       * Backdrop Sepia
       * @see https://tailwindcss.com/docs/backdrop-sepia
       */
      "backdrop-sepia": [{
        "backdrop-sepia": ["", g, u, d]
      }],
      // --------------
      // --- Tables ---
      // --------------
      /**
       * Border Collapse
       * @see https://tailwindcss.com/docs/border-collapse
       */
      "border-collapse": [{
        border: ["collapse", "separate"]
      }],
      /**
       * Border Spacing
       * @see https://tailwindcss.com/docs/border-spacing
       */
      "border-spacing": [{
        "border-spacing": p()
      }],
      /**
       * Border Spacing X
       * @see https://tailwindcss.com/docs/border-spacing
       */
      "border-spacing-x": [{
        "border-spacing-x": p()
      }],
      /**
       * Border Spacing Y
       * @see https://tailwindcss.com/docs/border-spacing
       */
      "border-spacing-y": [{
        "border-spacing-y": p()
      }],
      /**
       * Table Layout
       * @see https://tailwindcss.com/docs/table-layout
       */
      "table-layout": [{
        table: ["auto", "fixed"]
      }],
      /**
       * Caption Side
       * @see https://tailwindcss.com/docs/caption-side
       */
      caption: [{
        caption: ["top", "bottom"]
      }],
      // ---------------------------------
      // --- Transitions and Animation ---
      // ---------------------------------
      /**
       * Transition Property
       * @see https://tailwindcss.com/docs/transition-property
       */
      transition: [{
        transition: ["", "all", "colors", "opacity", "shadow", "transform", "none", u, d]
      }],
      /**
       * Transition Behavior
       * @see https://tailwindcss.com/docs/transition-behavior
       */
      "transition-behavior": [{
        transition: ["normal", "discrete"]
      }],
      /**
       * Transition Duration
       * @see https://tailwindcss.com/docs/transition-duration
       */
      duration: [{
        duration: [g, "initial", u, d]
      }],
      /**
       * Transition Timing Function
       * @see https://tailwindcss.com/docs/transition-timing-function
       */
      ease: [{
        ease: ["linear", "initial", H, u, d]
      }],
      /**
       * Transition Delay
       * @see https://tailwindcss.com/docs/transition-delay
       */
      delay: [{
        delay: [g, u, d]
      }],
      /**
       * Animation
       * @see https://tailwindcss.com/docs/animation
       */
      animate: [{
        animate: ["none", re, u, d]
      }],
      // ------------------
      // --- Transforms ---
      // ------------------
      /**
       * Backface Visibility
       * @see https://tailwindcss.com/docs/backface-visibility
       */
      backface: [{
        backface: ["hidden", "visible"]
      }],
      /**
       * Perspective
       * @see https://tailwindcss.com/docs/perspective
       */
      perspective: [{
        perspective: [S, u, d]
      }],
      /**
       * Perspective Origin
       * @see https://tailwindcss.com/docs/perspective-origin
       */
      "perspective-origin": [{
        "perspective-origin": V()
      }],
      /**
       * Rotate
       * @see https://tailwindcss.com/docs/rotate
       */
      rotate: [{
        rotate: $()
      }],
      /**
       * Rotate X
       * @see https://tailwindcss.com/docs/rotate
       */
      "rotate-x": [{
        "rotate-x": $()
      }],
      /**
       * Rotate Y
       * @see https://tailwindcss.com/docs/rotate
       */
      "rotate-y": [{
        "rotate-y": $()
      }],
      /**
       * Rotate Z
       * @see https://tailwindcss.com/docs/rotate
       */
      "rotate-z": [{
        "rotate-z": $()
      }],
      /**
       * Scale
       * @see https://tailwindcss.com/docs/scale
       */
      scale: [{
        scale: Y()
      }],
      /**
       * Scale X
       * @see https://tailwindcss.com/docs/scale
       */
      "scale-x": [{
        "scale-x": Y()
      }],
      /**
       * Scale Y
       * @see https://tailwindcss.com/docs/scale
       */
      "scale-y": [{
        "scale-y": Y()
      }],
      /**
       * Scale Z
       * @see https://tailwindcss.com/docs/scale
       */
      "scale-z": [{
        "scale-z": Y()
      }],
      /**
       * Scale 3D
       * @see https://tailwindcss.com/docs/scale
       */
      "scale-3d": ["scale-3d"],
      /**
       * Skew
       * @see https://tailwindcss.com/docs/skew
       */
      skew: [{
        skew: ie()
      }],
      /**
       * Skew X
       * @see https://tailwindcss.com/docs/skew
       */
      "skew-x": [{
        "skew-x": ie()
      }],
      /**
       * Skew Y
       * @see https://tailwindcss.com/docs/skew
       */
      "skew-y": [{
        "skew-y": ie()
      }],
      /**
       * Transform
       * @see https://tailwindcss.com/docs/transform
       */
      transform: [{
        transform: [u, d, "", "none", "gpu", "cpu"]
      }],
      /**
       * Transform Origin
       * @see https://tailwindcss.com/docs/transform-origin
       */
      "transform-origin": [{
        origin: V()
      }],
      /**
       * Transform Style
       * @see https://tailwindcss.com/docs/transform-style
       */
      "transform-style": [{
        transform: ["3d", "flat"]
      }],
      /**
       * Translate
       * @see https://tailwindcss.com/docs/translate
       */
      translate: [{
        translate: pe()
      }],
      /**
       * Translate X
       * @see https://tailwindcss.com/docs/translate
       */
      "translate-x": [{
        "translate-x": pe()
      }],
      /**
       * Translate Y
       * @see https://tailwindcss.com/docs/translate
       */
      "translate-y": [{
        "translate-y": pe()
      }],
      /**
       * Translate Z
       * @see https://tailwindcss.com/docs/translate
       */
      "translate-z": [{
        "translate-z": pe()
      }],
      /**
       * Translate None
       * @see https://tailwindcss.com/docs/translate
       */
      "translate-none": ["translate-none"],
      // ---------------------
      // --- Interactivity ---
      // ---------------------
      /**
       * Accent Color
       * @see https://tailwindcss.com/docs/accent-color
       */
      accent: [{
        accent: f()
      }],
      /**
       * Appearance
       * @see https://tailwindcss.com/docs/appearance
       */
      appearance: [{
        appearance: ["none", "auto"]
      }],
      /**
       * Caret Color
       * @see https://tailwindcss.com/docs/just-in-time-mode#caret-color-utilities
       */
      "caret-color": [{
        caret: f()
      }],
      /**
       * Color Scheme
       * @see https://tailwindcss.com/docs/color-scheme
       */
      "color-scheme": [{
        scheme: ["normal", "dark", "light", "light-dark", "only-dark", "only-light"]
      }],
      /**
       * Cursor
       * @see https://tailwindcss.com/docs/cursor
       */
      cursor: [{
        cursor: ["auto", "default", "pointer", "wait", "text", "move", "help", "not-allowed", "none", "context-menu", "progress", "cell", "crosshair", "vertical-text", "alias", "copy", "no-drop", "grab", "grabbing", "all-scroll", "col-resize", "row-resize", "n-resize", "e-resize", "s-resize", "w-resize", "ne-resize", "nw-resize", "se-resize", "sw-resize", "ew-resize", "ns-resize", "nesw-resize", "nwse-resize", "zoom-in", "zoom-out", u, d]
      }],
      /**
       * Field Sizing
       * @see https://tailwindcss.com/docs/field-sizing
       */
      "field-sizing": [{
        "field-sizing": ["fixed", "content"]
      }],
      /**
       * Pointer Events
       * @see https://tailwindcss.com/docs/pointer-events
       */
      "pointer-events": [{
        "pointer-events": ["auto", "none"]
      }],
      /**
       * Resize
       * @see https://tailwindcss.com/docs/resize
       */
      resize: [{
        resize: ["none", "", "y", "x"]
      }],
      /**
       * Scroll Behavior
       * @see https://tailwindcss.com/docs/scroll-behavior
       */
      "scroll-behavior": [{
        scroll: ["auto", "smooth"]
      }],
      /**
       * Scroll Margin
       * @see https://tailwindcss.com/docs/scroll-margin
       */
      "scroll-m": [{
        "scroll-m": p()
      }],
      /**
       * Scroll Margin X
       * @see https://tailwindcss.com/docs/scroll-margin
       */
      "scroll-mx": [{
        "scroll-mx": p()
      }],
      /**
       * Scroll Margin Y
       * @see https://tailwindcss.com/docs/scroll-margin
       */
      "scroll-my": [{
        "scroll-my": p()
      }],
      /**
       * Scroll Margin Start
       * @see https://tailwindcss.com/docs/scroll-margin
       */
      "scroll-ms": [{
        "scroll-ms": p()
      }],
      /**
       * Scroll Margin End
       * @see https://tailwindcss.com/docs/scroll-margin
       */
      "scroll-me": [{
        "scroll-me": p()
      }],
      /**
       * Scroll Margin Top
       * @see https://tailwindcss.com/docs/scroll-margin
       */
      "scroll-mt": [{
        "scroll-mt": p()
      }],
      /**
       * Scroll Margin Right
       * @see https://tailwindcss.com/docs/scroll-margin
       */
      "scroll-mr": [{
        "scroll-mr": p()
      }],
      /**
       * Scroll Margin Bottom
       * @see https://tailwindcss.com/docs/scroll-margin
       */
      "scroll-mb": [{
        "scroll-mb": p()
      }],
      /**
       * Scroll Margin Left
       * @see https://tailwindcss.com/docs/scroll-margin
       */
      "scroll-ml": [{
        "scroll-ml": p()
      }],
      /**
       * Scroll Padding
       * @see https://tailwindcss.com/docs/scroll-padding
       */
      "scroll-p": [{
        "scroll-p": p()
      }],
      /**
       * Scroll Padding X
       * @see https://tailwindcss.com/docs/scroll-padding
       */
      "scroll-px": [{
        "scroll-px": p()
      }],
      /**
       * Scroll Padding Y
       * @see https://tailwindcss.com/docs/scroll-padding
       */
      "scroll-py": [{
        "scroll-py": p()
      }],
      /**
       * Scroll Padding Start
       * @see https://tailwindcss.com/docs/scroll-padding
       */
      "scroll-ps": [{
        "scroll-ps": p()
      }],
      /**
       * Scroll Padding End
       * @see https://tailwindcss.com/docs/scroll-padding
       */
      "scroll-pe": [{
        "scroll-pe": p()
      }],
      /**
       * Scroll Padding Top
       * @see https://tailwindcss.com/docs/scroll-padding
       */
      "scroll-pt": [{
        "scroll-pt": p()
      }],
      /**
       * Scroll Padding Right
       * @see https://tailwindcss.com/docs/scroll-padding
       */
      "scroll-pr": [{
        "scroll-pr": p()
      }],
      /**
       * Scroll Padding Bottom
       * @see https://tailwindcss.com/docs/scroll-padding
       */
      "scroll-pb": [{
        "scroll-pb": p()
      }],
      /**
       * Scroll Padding Left
       * @see https://tailwindcss.com/docs/scroll-padding
       */
      "scroll-pl": [{
        "scroll-pl": p()
      }],
      /**
       * Scroll Snap Align
       * @see https://tailwindcss.com/docs/scroll-snap-align
       */
      "snap-align": [{
        snap: ["start", "end", "center", "align-none"]
      }],
      /**
       * Scroll Snap Stop
       * @see https://tailwindcss.com/docs/scroll-snap-stop
       */
      "snap-stop": [{
        snap: ["normal", "always"]
      }],
      /**
       * Scroll Snap Type
       * @see https://tailwindcss.com/docs/scroll-snap-type
       */
      "snap-type": [{
        snap: ["none", "x", "y", "both"]
      }],
      /**
       * Scroll Snap Type Strictness
       * @see https://tailwindcss.com/docs/scroll-snap-type
       */
      "snap-strictness": [{
        snap: ["mandatory", "proximity"]
      }],
      /**
       * Touch Action
       * @see https://tailwindcss.com/docs/touch-action
       */
      touch: [{
        touch: ["auto", "none", "manipulation"]
      }],
      /**
       * Touch Action X
       * @see https://tailwindcss.com/docs/touch-action
       */
      "touch-x": [{
        "touch-pan": ["x", "left", "right"]
      }],
      /**
       * Touch Action Y
       * @see https://tailwindcss.com/docs/touch-action
       */
      "touch-y": [{
        "touch-pan": ["y", "up", "down"]
      }],
      /**
       * Touch Action Pinch Zoom
       * @see https://tailwindcss.com/docs/touch-action
       */
      "touch-pz": ["touch-pinch-zoom"],
      /**
       * User Select
       * @see https://tailwindcss.com/docs/user-select
       */
      select: [{
        select: ["none", "text", "all", "auto"]
      }],
      /**
       * Will Change
       * @see https://tailwindcss.com/docs/will-change
       */
      "will-change": [{
        "will-change": ["auto", "scroll", "contents", "transform", u, d]
      }],
      // -----------
      // --- SVG ---
      // -----------
      /**
       * Fill
       * @see https://tailwindcss.com/docs/fill
       */
      fill: [{
        fill: ["none", ...f()]
      }],
      /**
       * Stroke Width
       * @see https://tailwindcss.com/docs/stroke-width
       */
      "stroke-w": [{
        stroke: [g, ue, Q, Re]
      }],
      /**
       * Stroke
       * @see https://tailwindcss.com/docs/stroke
       */
      stroke: [{
        stroke: ["none", ...f()]
      }],
      // ---------------------
      // --- Accessibility ---
      // ---------------------
      /**
       * Forced Color Adjust
       * @see https://tailwindcss.com/docs/forced-color-adjust
       */
      "forced-color-adjust": [{
        "forced-color-adjust": ["auto", "none"]
      }]
    },
    conflictingClassGroups: {
      overflow: ["overflow-x", "overflow-y"],
      overscroll: ["overscroll-x", "overscroll-y"],
      inset: ["inset-x", "inset-y", "start", "end", "top", "right", "bottom", "left"],
      "inset-x": ["right", "left"],
      "inset-y": ["top", "bottom"],
      flex: ["basis", "grow", "shrink"],
      gap: ["gap-x", "gap-y"],
      p: ["px", "py", "ps", "pe", "pt", "pr", "pb", "pl"],
      px: ["pr", "pl"],
      py: ["pt", "pb"],
      m: ["mx", "my", "ms", "me", "mt", "mr", "mb", "ml"],
      mx: ["mr", "ml"],
      my: ["mt", "mb"],
      size: ["w", "h"],
      "font-size": ["leading"],
      "fvn-normal": ["fvn-ordinal", "fvn-slashed-zero", "fvn-figure", "fvn-spacing", "fvn-fraction"],
      "fvn-ordinal": ["fvn-normal"],
      "fvn-slashed-zero": ["fvn-normal"],
      "fvn-figure": ["fvn-normal"],
      "fvn-spacing": ["fvn-normal"],
      "fvn-fraction": ["fvn-normal"],
      "line-clamp": ["display", "overflow"],
      rounded: ["rounded-s", "rounded-e", "rounded-t", "rounded-r", "rounded-b", "rounded-l", "rounded-ss", "rounded-se", "rounded-ee", "rounded-es", "rounded-tl", "rounded-tr", "rounded-br", "rounded-bl"],
      "rounded-s": ["rounded-ss", "rounded-es"],
      "rounded-e": ["rounded-se", "rounded-ee"],
      "rounded-t": ["rounded-tl", "rounded-tr"],
      "rounded-r": ["rounded-tr", "rounded-br"],
      "rounded-b": ["rounded-br", "rounded-bl"],
      "rounded-l": ["rounded-tl", "rounded-bl"],
      "border-spacing": ["border-spacing-x", "border-spacing-y"],
      "border-w": ["border-w-x", "border-w-y", "border-w-s", "border-w-e", "border-w-t", "border-w-r", "border-w-b", "border-w-l"],
      "border-w-x": ["border-w-r", "border-w-l"],
      "border-w-y": ["border-w-t", "border-w-b"],
      "border-color": ["border-color-x", "border-color-y", "border-color-s", "border-color-e", "border-color-t", "border-color-r", "border-color-b", "border-color-l"],
      "border-color-x": ["border-color-r", "border-color-l"],
      "border-color-y": ["border-color-t", "border-color-b"],
      translate: ["translate-x", "translate-y", "translate-none"],
      "translate-none": ["translate", "translate-x", "translate-y", "translate-z"],
      "scroll-m": ["scroll-mx", "scroll-my", "scroll-ms", "scroll-me", "scroll-mt", "scroll-mr", "scroll-mb", "scroll-ml"],
      "scroll-mx": ["scroll-mr", "scroll-ml"],
      "scroll-my": ["scroll-mt", "scroll-mb"],
      "scroll-p": ["scroll-px", "scroll-py", "scroll-ps", "scroll-pe", "scroll-pt", "scroll-pr", "scroll-pb", "scroll-pl"],
      "scroll-px": ["scroll-pr", "scroll-pl"],
      "scroll-py": ["scroll-pt", "scroll-pb"],
      touch: ["touch-x", "touch-y", "touch-pz"],
      "touch-x": ["touch"],
      "touch-y": ["touch"],
      "touch-pz": ["touch"]
    },
    conflictingClassGroupModifiers: {
      "font-size": ["leading"]
    },
    orderSensitiveModifiers: ["*", "**", "after", "backdrop", "before", "details-content", "file", "first-letter", "first-line", "marker", "placeholder", "selection"]
  };
}, Wr = /* @__PURE__ */ Cr(Ur), Br = (e, r, t) => Math.min(Math.max(e, r), t), xe = (e) => Math.round(Br(e, 0, 255)), ze = (e) => {
  const r = e.replace("#", ""), t = parseInt(r, 16);
  return {
    r: t >> 16 & 255,
    g: t >> 8 & 255,
    b: t & 255
  };
}, Ne = ({ r: e, g: r, b: t }) => {
  const o = (n) => xe(n).toString(16).padStart(2, "0");
  return `#${o(e)}${o(r)}${o(t)}`;
}, Me = ({ h: e, s: r, v: t }) => {
  e = e / 360, r = r / 100, t = t / 100;
  const o = Math.floor(e * 6), n = e * 6 - o, a = t * (1 - r), s = t * (1 - n * r), c = t * (1 - (1 - n) * r);
  let i, m, b;
  switch (o % 6) {
    case 0:
      i = t, m = c, b = a;
      break;
    case 1:
      i = s, m = t, b = a;
      break;
    case 2:
      i = a, m = t, b = c;
      break;
    case 3:
      i = a, m = s, b = t;
      break;
    case 4:
      i = c, m = a, b = t;
      break;
    case 5:
      i = t, m = a, b = s;
      break;
    default:
      i = m = b = 0;
  }
  return {
    r: xe(i * 255),
    g: xe(m * 255),
    b: xe(b * 255)
  };
}, Ae = ({ r: e, g: r, b: t }) => {
  e /= 255, r /= 255, t /= 255;
  const o = Math.max(e, r, t), n = Math.min(e, r, t), a = o - n;
  let s = 0;
  const c = o === 0 ? 0 : a / o, i = o;
  if (a !== 0)
    switch (o) {
      case e:
        s = ((r - t) / a + (r < t ? 6 : 0)) / 6;
        break;
      case r:
        s = ((t - e) / a + 2) / 6;
        break;
      case t:
        s = ((e - r) / a + 4) / 6;
        break;
    }
  return {
    h: Math.round(s * 360),
    s: Math.round(c * 100),
    v: Math.round(i * 100)
  };
}, qr = (e) => Ne(Me(e)), Jr = (e) => Ae(ze(e)), Xr = ({ r: e, g: r, b: t }) => {
  e /= 255, r /= 255, t /= 255;
  const o = Math.max(e, r, t), n = Math.min(e, r, t), a = o - n;
  let s = 0, c = 0;
  const i = (o + n) / 2;
  if (a !== 0)
    switch (c = i > 0.5 ? a / (2 - o - n) : a / (o + n), o) {
      case e:
        s = ((r - t) / a + (r < t ? 6 : 0)) / 6;
        break;
      case r:
        s = ((t - e) / a + 2) / 6;
        break;
      case t:
        s = ((e - r) / a + 4) / 6;
        break;
    }
  return {
    h: Math.round(s * 360),
    s: Math.round(c * 100),
    l: Math.round(i * 100)
  };
}, Kr = ({ h: e, s: r, l: t }) => {
  e /= 360, r /= 100, t /= 100;
  const o = (c, i, m) => (m < 0 && (m += 1), m > 1 && (m -= 1), m < 1 / 6 ? c + (i - c) * 6 * m : m < 1 / 2 ? i : m < 2 / 3 ? c + (i - c) * (2 / 3 - m) * 6 : c);
  let n, a, s;
  if (r === 0)
    n = a = s = t;
  else {
    const c = t < 0.5 ? t * (1 + r) : t + r - t * r, i = 2 * t - c;
    n = o(i, c, e + 1 / 3), a = o(i, c, e), s = o(i, c, e - 1 / 3);
  }
  return {
    r: Math.round(n * 255),
    g: Math.round(a * 255),
    b: Math.round(s * 255)
  };
}, Zr = (e) => Ne(Kr(e)), ht = (e) => Xr(ze(e)), Ye = ({ h: e, s: r, v: t }) => {
  const o = t * (2 - r / 100) / 2, n = o !== 0 && o !== 100 ? (t - o) / Math.min(o, 100 - o) * 100 : 0;
  return {
    h: e,
    s: Math.round(n),
    l: Math.round(o)
  };
}, Ue = ({ h: e, s: r, l: t }) => {
  const o = t + r * Math.min(t, 100 - t) / 100, n = o === 0 ? 0 : 2 * (1 - t / o) * 100;
  return {
    h: e,
    s: Math.round(n),
    v: Math.round(o)
  };
}, Qr = (e) => {
  const { r, g: t, b: o } = ze(e);
  return (0.299 * r + 0.587 * t + 0.114 * o) / 255 > 0.5 ? "#000000" : "#ffffff";
}, gt = () => `#${Math.floor(Math.random() * 16777215).toString(16).padStart(6, "0")}`, et = (e, r = "hex") => {
  switch (r) {
    case "hex":
      return e.getHex();
    case "rgb":
      const t = e.getRgb();
      return `rgb(${t.r}, ${t.g}, ${t.b})`;
    case "rgba":
      const o = e.getRgba();
      return `rgba(${o.r}, ${o.g}, ${o.b}, ${o.a})`;
    case "hsl":
      const n = e.getHsl();
      return `hsl(${n.h}, ${n.s}%, ${n.l}%)`;
    case "hsla":
      const a = e.getHsla();
      return `hsla(${a.h}, ${a.s}%, ${a.l}%, ${a.a})`;
    default:
      ye();
  }
};
function rt(e) {
  switch (e.type) {
    case "hex":
      return { ...Jr(e.value), a: 1 };
    case "rgb":
      return { ...Ae(e), a: 1 };
    case "rgba":
      return { ...Ae(e), a: e.a };
    case "hsl":
      return { ...Ue(e), a: 1 };
    case "hsla":
      return { ...Ue(e), a: 1 };
    case "hsv":
      return { ...e, a: 1 };
    case "hsva":
      return { ...e };
    default:
      ye();
  }
}
function W(e, r) {
  const t = rt(e);
  switch (r) {
    case "hex":
      return qr(t);
    case "rgb":
      return Me(t);
    case "rgba":
      return { ...Me(t), a: t.a };
    case "hsl":
      return Ye(t);
    case "hsla":
      return { ...Ye(t), a: t.a };
    case "hsv":
      return { h: t.h, s: t.s, v: t.v };
    case "hsva":
      return t;
    default:
      ye();
  }
}
const ee = (...e) => Wr(mr(e)), F = (e, r, t) => Math.min(Math.max(e, r), t), B = (e) => Math.round(e), tt = (e) => {
  const r = e.replace("#", "");
  return /^([0-9A-Fa-f]{3}|[0-9A-Fa-f]{6}|[0-9A-Fa-f]{8})$/.test(r);
}, ot = (e) => {
  const r = e.replace("#", "");
  return r.length === 3 ? `#${r[0]}${r[0]}${r[1]}${r[1]}${r[2]}${r[2]}` : `#${r}`;
}, nt = (e) => {
  const r = e.replace("#", "");
  return r.length === 8 ? `#${r.substring(0, 6)}` : e;
}, bt = (e) => {
  const r = e.trim();
  if (r.startsWith("#"))
    return tt(r) ? nt(ot(r)) : "#ff0000";
  const t = r.match(/rgba?\((\d+),\s*(\d+),\s*(\d+)(?:,\s*[\d.]+)?\)/);
  if (t) {
    const s = parseInt(t[1]), c = parseInt(t[2]), i = parseInt(t[3]);
    if (s <= 255 && c <= 255 && i <= 255)
      return Ne({ r: s, g: c, b: i });
  }
  const o = r.match(/hsla?\((\d+),\s*(\d+)%,\s*(\d+)%(?:,\s*[\d.]+)?\)/);
  if (o) {
    const s = parseInt(o[1]), c = parseInt(o[2]), i = parseInt(o[3]);
    if (s <= 360 && c <= 100 && i <= 100)
      return Zr({ h: s, s: c, l: i });
  }
  const n = {
    red: "#ff0000",
    green: "#008000",
    blue: "#0000ff",
    white: "#ffffff",
    black: "#000000",
    yellow: "#ffff00",
    cyan: "#00ffff",
    magenta: "#ff00ff",
    gray: "#808080",
    orange: "#ffa500",
    purple: "#800080",
    pink: "#ffc0cb"
  }, a = r.toLowerCase();
  return n[a] ? n[a] : "#ff0000";
};
function ye(e) {
  throw new Error("You should never get here.");
}
class q {
  constructor(r) {
    this.colorInput = r, typeof r == "string" ? this.color = { type: "hex", value: r } : "type" in r ? this.color = r : "r" in r ? "a" in r ? this.color = { type: "rgba", ...r } : this.color = { type: "rgb", ...r } : "v" in r ? "a" in r ? this.color = { type: "hsva", ...r } : this.color = { type: "hsv", ...r } : "a" in r ? this.color = { type: "hsla", ...r } : this.color = { type: "hsl", ...r };
  }
  getRgb() {
    return W(this.color, "rgb");
  }
  getHsv() {
    return W(this.color, "hsv");
  }
  getHsla() {
    return W(this.color, "hsla");
  }
  getHsva() {
    return W(this.color, "hsva");
  }
  getHex() {
    return W(this.color, "hex");
  }
  getHsl() {
    return W(this.color, "hsl");
  }
  getRgba() {
    return W(this.color, "rgba");
  }
  format(r = "hex") {
    return et(this, r);
  }
  getContrastingColor() {
    return new q({ type: "hex", value: Qr(this.getHex()) });
  }
}
const We = (e, r) => {
  const t = e.getBoundingClientRect();
  let o, n;
  if ("touches" in r) {
    const m = r.touches[0] || r.changedTouches[0];
    o = m.clientX, n = m.clientY;
  } else
    o = r.clientX, n = r.clientY;
  const a = o - t.left, s = n - t.top, c = F(a / t.width, 0, 1), i = F(s / t.height, 0, 1);
  return { left: c, top: i };
}, $e = ({ onMove: e, onMoveEnd: r, onKey: t, children: o, className: n, ...a }) => {
  const s = ve(null), c = ve(!1), i = A(
    (w) => {
      !c.current || !s.current || (w.preventDefault(), e(We(s.current, w)));
    },
    [e]
  ), m = A(() => {
    c.current = !1, r(), document.removeEventListener("mousemove", i), document.removeEventListener("mouseup", m), document.removeEventListener("touchmove", i), document.removeEventListener("touchend", m);
  }, [i, r]), b = A(
    (w) => {
      const R = s.current;
      R && (w.preventDefault(), c.current = !0, e(We(R, w.nativeEvent)), document.addEventListener("mousemove", i), document.addEventListener("mouseup", m), document.addEventListener("touchmove", i, { passive: !1 }), document.addEventListener("touchend", m));
    },
    [e, i, m]
  ), P = A(
    (w) => {
      if (!t) return;
      const R = w.which || w.keyCode;
      R < 37 || R > 40 || (w.preventDefault(), t({
        left: R === 39 ? 0.05 : R === 37 ? -0.05 : 0,
        top: R === 40 ? 0.05 : R === 38 ? -0.05 : 0
      }));
    },
    [t]
  );
  return Pe(() => () => {
    document.removeEventListener("mousemove", i), document.removeEventListener("mouseup", m), document.removeEventListener("touchmove", i), document.removeEventListener("touchend", m);
  }, [i, m]), /* @__PURE__ */ E.jsx(
    "div",
    {
      ref: s,
      className: ee("rounded-inherit absolute inset-0 touch-none outline-none", "focus:[&_.react-colorful-pointer]:scale-110", n),
      onTouchStart: b,
      onMouseDown: b,
      onKeyDown: P,
      tabIndex: 0,
      role: "slider",
      ...a,
      children: o
    }
  );
}, Oe = ({ className: e, top: r = 0.5, left: t, color: o }) => /* @__PURE__ */ E.jsx(
  "div",
  {
    className: ee(
      "absolute z-[1] h-6 w-6 -translate-x-1/2 -translate-y-1/2 cursor-grab rounded-full border-2 border-white bg-white active:cursor-grabbing",
      "react-colorful-pointer shadow-[0_2px_8px_rgba(0,0,0,0.15)] transition-transform",
      e
    ),
    style: {
      top: `${r * 100}%`,
      left: `${t * 100}%`
    }
  }
), st = ({ hsva: e, onChange: r, className: t, onFinishedUpdates: o }) => {
  const n = A(
    (i) => {
      r({ a: i.left }, !1);
    },
    [r]
  ), a = A(
    (i) => {
      r(
        {
          a: F(e.a + i.left, 0, 1)
        },
        !0
      );
    },
    [e.a, r]
  ), s = we(() => new q({ type: "hsva", h: e.h, s: e.s, v: e.v, a: 0 }).format("hsla"), [e.h, e.s, e.v]), c = we(() => new q({ type: "hsva", h: e.h, s: e.s, v: e.v, a: 1 }).format("hsla"), [e.h, e.s, e.v]);
  return /* @__PURE__ */ E.jsx("div", { className: ee("relative h-full w-full", t), children: /* @__PURE__ */ E.jsxs(
    $e,
    {
      onMove: n,
      onMoveEnd: o,
      onKey: a,
      "aria-label": "Alpha",
      "aria-valuetext": `${B(e.a * 100)}%`,
      "aria-valuenow": B(e.a * 100),
      "aria-valuemin": 0,
      "aria-valuemax": 100,
      className: "h-full w-full rounded-lg",
      children: [
        /* @__PURE__ */ E.jsx(
          "div",
          {
            className: "absolute inset-0 rounded-lg bg-white",
            style: {
              backgroundImage: `url('data:image/svg+xml,<svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill-opacity=".05"><rect x="8" width="8" height="8"/><rect y="8" width="8" height="8"/></svg>')`
            }
          }
        ),
        /* @__PURE__ */ E.jsx(
          "div",
          {
            className: "absolute inset-0 rounded-lg",
            style: {
              backgroundImage: `linear-gradient(90deg, ${s}, ${c})`
            }
          }
        ),
        /* @__PURE__ */ E.jsx(
          Oe,
          {
            className: "z-[1]",
            top: 0.5,
            left: e.a,
            color: ""
          }
        )
      ]
    }
  ) });
}, at = ({ hue: e, onChange: r, className: t, onFinishedUpdates: o }) => {
  const n = A(
    (s) => {
      const c = B(F(360 * s.left, 0, 360));
      r({ h: c }, !1);
    },
    [r]
  ), a = A(
    (s) => {
      r(
        {
          h: F(e + s.left * 360, 0, 360)
        },
        !0
      );
    },
    [e, r]
  );
  return /* @__PURE__ */ E.jsx("div", { className: ee("relative h-full w-full", t), children: /* @__PURE__ */ E.jsxs(
    $e,
    {
      onMove: n,
      onMoveEnd: o,
      onKey: a,
      "aria-label": "Hue",
      "aria-valuenow": B(e),
      "aria-valuemax": 360,
      "aria-valuemin": 0,
      className: "h-full w-full rounded-lg",
      children: [
        /* @__PURE__ */ E.jsx(
          "div",
          {
            className: "absolute inset-0 rounded-lg",
            style: {
              background: "linear-gradient(to right, #f00 0%, #ff0 17%, #0f0 33%, #0ff 50%, #00f 67%, #f0f 83%, #f00 100%)"
            }
          }
        ),
        /* @__PURE__ */ E.jsx(
          Oe,
          {
            className: "z-[2]",
            top: 0.5,
            left: e / 360,
            color: ""
          }
        )
      ]
    }
  ) });
}, lt = ({ hsva: e, onChange: r, className: t, onFinishedUpdates: o }) => {
  const n = A(
    (c) => {
      const i = B(F(c.left * 100, 0, 100)), m = B(F(100 - c.top * 100, 0, 100));
      r({ s: i, v: m }, !1);
    },
    [r]
  ), a = A(
    (c) => {
      r(
        {
          s: F(e.s + c.left * 100, 0, 100),
          v: F(e.v - c.top * 100, 0, 100)
        },
        !0
      );
    },
    [e.s, e.v, r]
  ), s = `hsl(${e.h}, 100%, 50%)`;
  return /* @__PURE__ */ E.jsx("div", { className: ee("relative h-full w-full rounded-xl", t), children: /* @__PURE__ */ E.jsxs(
    $e,
    {
      onMove: n,
      onMoveEnd: o,
      onKey: a,
      "aria-label": "Color",
      "aria-valuetext": `Saturation ${B(e.s)}%, Brightness ${B(e.v)}%`,
      className: "rounded-inherit h-full w-full",
      children: [
        /* @__PURE__ */ E.jsx(
          "div",
          {
            className: "rounded-inherit absolute inset-0 rounded-b-none",
            style: {
              backgroundColor: s,
              backgroundImage: "linear-gradient(to top, #000, rgba(0, 0, 0, 0)), linear-gradient(to right, #fff, rgba(255, 255, 255, 0))",
              boxShadow: "inset 0 0 0 1px rgba(0, 0, 0, 0.05)"
            }
          }
        ),
        /* @__PURE__ */ E.jsx(
          Oe,
          {
            className: "z-[3]",
            top: 1 - e.v / 100,
            left: e.s / 100,
            color: ""
          }
        )
      ]
    }
  ) });
}, sr = lr(null), ke = () => {
  const e = ir(sr);
  if (!e)
    throw new Error("ColorPicker compound components must be used within a ColorPicker");
  return e;
};
function Be(e) {
  return "type" in e ? new q(e) : e;
}
const it = ({ color: e, onChange: r, className: t, children: o, defaultColor: n, ...a }) => {
  const [s, c] = _e(Be(e ?? n ?? { h: 340, s: 58, v: 100, a: 1, type: "hsva" })), i = ve(!1), m = ve(s);
  Pe(() => {
    !i.current && e && (m.current = Be(e), c(m.current));
  }, [e]);
  const b = A(
    (C, S) => {
      const j = new q({ type: "hsva", ...m.current.getHsva(), ...C });
      m.current = j, c(j), i.current = !S, r?.(j);
    },
    [r]
  ), P = A(() => {
    i.current = !1;
  }, []), w = A(async () => {
    if (!("EyeDropper" in window)) {
      console.log("EyeDropper API not supported");
      return;
    }
    try {
      const S = await new window.EyeDropper().open();
      S.sRGBHex && b(W({ type: "hex", value: S.sRGBHex }, "hsva"), !0);
    } catch (C) {
      console.log("EyeDropper cancelled or failed:", C);
    }
  }, [r]), R = {
    hsva: s.getHsva(),
    updateHsva: b,
    handleEyeDropper: w,
    onFinishedUpdates: P
  };
  return /* @__PURE__ */ E.jsx(sr.Provider, { value: R, children: /* @__PURE__ */ E.jsx(
    "div",
    {
      ...a,
      className: ee("relative flex h-[280px] w-[280px] cursor-default flex-col rounded-2xl bg-white shadow-lg select-none", t),
      children: o
    }
  ) });
}, ct = ({ className: e }) => {
  const { hsva: r, updateHsva: t, onFinishedUpdates: o } = ke(), n = A(
    (a, s) => {
      t(a, s);
    },
    [t]
  );
  return /* @__PURE__ */ E.jsx(
    lt,
    {
      hsva: r,
      onChange: n,
      className: e,
      onFinishedUpdates: o
    }
  );
}, dt = ({ className: e }) => {
  const { hsva: r, updateHsva: t, onFinishedUpdates: o } = ke(), n = A(
    (a, s) => {
      t(a, s);
    },
    [t]
  );
  return /* @__PURE__ */ E.jsx(
    at,
    {
      hue: r.h,
      onChange: n,
      className: e,
      onFinishedUpdates: o
    }
  );
}, ut = ({ className: e }) => {
  const { hsva: r, updateHsva: t, onFinishedUpdates: o } = ke(), n = A(
    (a, s) => {
      t(a, s);
    },
    [t]
  );
  return /* @__PURE__ */ E.jsx(
    st,
    {
      hsva: r,
      onChange: n,
      className: e,
      onFinishedUpdates: o
    }
  );
}, mt = ({ className: e, title: r = "Pick color from screen", children: t }) => {
  const { handleEyeDropper: o } = ke(), [n, a] = _e(!1);
  return Pe(() => {
    typeof window < "u" && "EyeDropper" in window && a(!0);
  }, []), n ? /* @__PURE__ */ E.jsx(
    "button",
    {
      onClick: o,
      className: ee("flex h-10 w-10 flex-shrink-0 cursor-pointer items-center justify-center rounded-lg transition-colors hover:bg-gray-100", e),
      title: r,
      children: t
    }
  ) : null;
}, xt = Object.assign(it, {
  Saturation: ct,
  Hue: dt,
  Alpha: ut,
  EyeDropper: mt
});
function ft(e, r) {
  switch (r) {
    case "hex":
      return { type: "hex", value: e.getHex() };
    case "rgb":
      return { type: "rgb", ...e.getRgb() };
    case "rgba":
      return { type: "rgba", ...e.getRgba() };
    case "hsl":
      return { type: "hsl", ...e.getHsl() };
    case "hsla":
      return { type: "hsla", ...e.getHsla() };
    case "hsv":
      return { type: "hsv", ...e.getHsv() };
    case "hsva":
      return { type: "hsva", ...e.getHsva() };
    default:
      ye();
  }
}
const vt = (e = { type: "hex", value: "#ff6b9d" }) => {
  const [r, t] = _e(() => {
    const s = typeof e == "function" ? e() : e;
    return { color: new q(s), type: s.type };
  }), o = we(() => ft(r.color, r.type), [r.color, r.type]), n = we(() => {
    const s = r.color.getHsva();
    return {
      hex: r.color.getHex(),
      rgb: r.color.getRgb(),
      rgba: r.color.getRgba(),
      hsl: r.color.getHsl(),
      hsla: r.color.getHsla(),
      hsv: r.color.getHsv(),
      hsva: s,
      alpha: s.a
    };
  }, [r.color]), a = A((s) => {
    try {
      const c = "type" in s ? new q(s) : s;
      t((i) => ({ color: c, type: i.type }));
    } catch (c) {
      console.warn("Invalid color input:", s, c);
    }
  }, []);
  return [
    {
      colorInput: o,
      colorState: n
    },
    a
  ];
}, wt = "2.0.0";
export {
  q as Color,
  xt as ColorPicker,
  ee as cn,
  W as convertColor,
  et as formatColorString,
  Qr as getContrastColor,
  ht as hexToHsl,
  Jr as hexToHsv,
  ze as hexToRgb,
  Zr as hslToHex,
  Ue as hslToHsv,
  Kr as hslToRgb,
  qr as hsvToHex,
  Ye as hsvToHsl,
  Me as hsvToRgb,
  bt as parseColorString,
  gt as randomHex,
  Ne as rgbToHex,
  Xr as rgbToHsl,
  Ae as rgbToHsv,
  vt as useColorState,
  wt as version
};
